<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>ontoProc: Ontology interfaces for Bioconductor, with focus on cell type identification • ontoProc</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ontoProc: Ontology interfaces for Bioconductor, with focus on cell type identification">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ontoProc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.99.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/ontoProc.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/owlents.html">owlents: using OWL directly in ontoProc</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/vjcitn/ontoProc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="ontoProc_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="ontoProc_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="ontoProc_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="ontoProc_files/datatables-binding-0.33/datatables.js"></script><link href="ontoProc_files/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="ontoProc_files/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="ontoProc_files/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="ontoProc_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="ontoProc_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>ontoProc: Ontology interfaces for Bioconductor, with focus on cell type identification</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">October 13, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/ontoProc/blob/HEAD/vignettes/ontoProc.Rmd"><code>vignettes/ontoProc.Rmd</code></a></small>
      <div class="d-none name"><code>ontoProc.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The ambitions of collaborative single cell biology will only be achieved through the coordinated efforts of many groups, to help clarify cell types and dynamics in an array of functional and environmental contexts. The use of formal ontology in this pursuit is well-motivated and research progress has already been substantial.</p>
<p><span class="citation">Bakken et al. (2017)</span> discuss “strategies for standardized cell type representations based on the data outputs from [high-content flow cytometry and single cell RNA sequencing], including ‘context annotations’ in the form of standardized experiment metadata about the specimen source analyzed and marker genes that serve as the most useful features in machine learning-based cell type classification models.” <span class="citation">Aevermann et al. (2018)</span> describe how the FAIR principles can be implemented using statistical identification of necessary and sufficient conditions for determining cell class membership. They propose that Cell Ontology can be transformed to a broadly usable knowledgebase through the incorporation of accurate marker gene signatures for cell classes.</p>
<p>In this vignette, we review key concepts and tasks required to make progress in the adoption and application of ontological discipline in Bioconductor-oriented data analysis.</p>
<p>We’ll start by setting up some package attachments and ontology objects.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn/ontoProc">ontoProc</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyPlot</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocStyle" class="external-link">BiocStyle</a></span><span class="op">)</span>  <span class="co"># for package references</span></span>
<span><span class="va">cl</span> <span class="op">=</span> <span class="fu"><a href="../reference/getOnto.html">getOnto</a></span><span class="op">(</span><span class="st">"cellOnto"</span>, <span class="st">"2021"</span><span class="op">)</span> <span class="co"># for continuity --    has_high_plasma_membrane_amount: list</span></span>
<span><span class="va">go</span> <span class="op">=</span> <span class="fu"><a href="../reference/getOnto.html">getOnto</a></span><span class="op">(</span><span class="st">"goOnto"</span>, <span class="st">"2021"</span><span class="op">)</span>  <span class="co"># if updated, some assertions will fail...</span></span>
<span><span class="va">pr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getOnto.html">getOnto</a></span><span class="op">(</span><span class="st">"Pronto"</span>, <span class="st">"2021"</span><span class="op">)</span>  <span class="co"># important case change</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="scope-of-package">Scope of package<a class="anchor" aria-label="anchor" href="#scope-of-package"></a>
</h2>
<div class="section level3">
<h3 id="owl-interface">OWL interface<a class="anchor" aria-label="anchor" href="#owl-interface"></a>
</h3>
<p>As of 1.99.0, facilities are present to import any valid OWL ontology. We use basilisk to incorporate functionality from <a href="https://owlready2.readthedocs.io/en/v0.47/" class="external-link">owlready2</a> and <a href="https://bioregistry.io/" class="external-link">bioregisty</a>. One way of identifying a large number of ontologies available for ingestion is to query bioregistry.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">br</span> <span class="op">=</span> <span class="fu"><a href="../reference/bioregistry_ols_resources.html">bioregistry_ols_resources</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">br</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291"],["Alzheimer's Disease Ontology is a knowledge-based ontology that encompasses varieties of concepts related to Alzheimer'S Disease, foundamentally structured by upper level Basic Formal Ontology(BFO). This Ontology is enriched by the interrelational entities that demonstrate the nextwork of the understanding on Alzheimer's disease and can be readily applied for text mining.","AEO is an ontology of anatomical structures that expands CARO, the Common Anatomy Reference Ontology","The Adverse Event Reporting Ontology (AERO) is an ontology aimed at supporting clinicians at the time of data entry, increasing quality and accuracy of reported adverse events","Allotrope Merged Ontology Suite","AgrO is an ontlogy for representing agronomic practices, techniques, variables and related entities","The ontology for the Anatomy of the Insect SkeletoMuscular system (AISM) contains terms used to describe the cuticle - as a single anatomical structure - and the skeletal muscle system, to be used in insect biodiversity research.","An ontology for the development and anatomy of Amphioxus (Branchiostoma lanceolatum).","A structured controlled vocabulary for the phenotypes of Ascomycete fungi.","An OWL2 ontology of phenomena in infectious disease epidemiology and population biology for use in epidemic simulation.","Antibiotic resistance genes and mutations","ATOL (Animal Trait Ontology for Livestock) is an ontology of characteristics defining phenotypes of livestock in their environment (EOL). ATOL aims to: - provide a reference ontology of phenotypic traits of farm animals for the international scientificand educational - communities, farmers, etc.; - deliver this reference ontology in a language which can be used by computers in order to support database management, semantic analysis and modeling; - represent traits as generic as possible for livestock vertebrates; - make the ATOL ontology as operational as possible and closely related to measurement techniques; - structure the ontology in relation to animal production.","The BioAssay Ontology (BAO) describes biological screening assays and their results including high-throughput screening (HTS) data for the purpose of categorizing assays and data analysis. BAO is an extensible, knowledge-based, highly expressive (currently SHOIQ(D)) description of biological assays making use of descriptive logic based features of the Web Ontology Language (OWL). BAO currently has over 700 classes and also makes use of several other ontologies. It describes several concepts related to biological screening, including Perturbagen, Format, Meta Target, Design, Detection Technology, and Endpoint. Perturbagens are perturbing agents that are screened in an assay; they are mostly small molecules. Assay Meta Target describes what is known about the biological system and / or its components interrogated in the assay (and influenced by the Perturbagen). Meta target can be directly described as a molecular entity (e.g. a purified protein or a protein complex), or indirectly by a biological process or event (e.g. phosphorylation). Format describes the biological or chemical features common to each test condition in the assay and includes biochemical, cell-based, organism-based, and variations thereof. The assay Design describes the assay methodology and implementation of how the perturbation of the biological system is translated into a detectable signal. Detection Technology relates to the physical method and technical details to detect and record a signal. Endpoints are the final HTS results as they are usually published (such as IC50, percent inhibition, etc). BAO has been designed to accommodate multiplexed assays. All main BAO components include multiple levels of sub-categories and specification classes, which are linked via object property relationships forming an expressive knowledge-based representation.","An application ontology built for beta cell genomics studies.","The Behaviour Change Intervention Ontology is an ontology for all aspects of human behaviour change interventions and their evaluation.","An ontology to support the interoperability of biodiversity data, including data on museum collections, environmental/metagenomic samples, and ecological surveys.","The upper level ontology upon which OBO Foundry ontologies are built.",null,"A classification of algorithms for simulating biology, their parameters, and their outputs","TEDDY is an ontology for dynamical behaviours, observable dynamical phenomena, and control elements of bio-models and biological systems in Systems and Synthetic Biology.","An ontology for respresenting spatial concepts, anatomical axes, gradients, regions, planes, sides and surfaces. These concepts can be used at multiple biological scales and in a diversity of taxa, including plants, animals and fungi. The BSPO is used to provide a source of anatomical location descriptors for logically defining anatomical entity classes in anatomy ontologies.","A structured controlled vocabulary for the source of an enzyme comprising tissues, cell lines, cell types and cell cultures.","The Chemical Analysis Ontology (CAO) is an ontology developed as part of the Chemical Analaysis Metadata Project (ChAMP) as a resource to semantically annotate standards developed using the ChAMP platform.  For more information go to http://champ-project.org.","The Common Anatomy Reference Ontology (CARO) is being developed to facilitate interoperability between existing anatomy ontologies for different species, and will provide a template for building new anatomy ontologies.","The Common Coordinate Framework (CCF) Ontology is an application ontology built to support the development of the Human Reference Atlas (HRA).  It unifies vocabulary for HRA construction and usage—making it possible to ingest external data sources; supporting uniform tissue sample registration that includes the spatial positioning and semantic annotations within 3D reference organs; and supporting user-formulated cross-domain queries over tissue donor properties, anatomical structures, cell types, biomarkers, and 3D space. The CCF Ontology consists of three major ontologies. The Biological Structure Ontology records anatomical structures, cell types, and biomarkers (ASCT+B) and the relationships between them.  The ASCT+B tables are authored by human experts using templated Google Sheets. The biomarkers, cell types, and anatomical structures are mapped to existing ontologies (Uberon/FMA, CL, HGNC) whenever possible.  All relationships between anatomical structures and from cell types to anatomical structures are valid Uberon and CL relationships. The Spatial Ontology defines the shape, size, location, and rotation of experimental tissue and data major anatomical structures in the 3D Reference Object Library. The Specimen Ontology captures the sex, age, and other information on donors that provided tissue data used in the construction of the HRA.","The Cell Cycle Ontology extends existing ontologies for cell cycle knowledge building a resource that integrates and manages knowledge about the cell cycle components and regulatory aspects.","a formalization of concepts and relations relevant to evolutionary comparative analysis","The CDNO provides structured terminologies to describe nutritional attributes of material entities that contribute to human diet. These terms are intended primarily to be associated with datasets that quantify concentration of chemical nutritional components derived from samples taken from any stage in the production of food raw materials (including from crops, livestock, fisheries) and through processing and supply chains. Additional knowledge associated with these dietary sources may be represented by terms that describe functional, physical and other attributes. Whilst recognising that dietary nutrients within food substrates may be present as complex and dynamic physical and chemical structures or mixtures, CDNO focuses on components typically quantified in an analytical chemistry laboratory. The primary CDNO class ‘dietary nutritional component’ contains hierarchical sets of terms organised to reflect commonly used classifications of chemical food composition. This class does not represent an exhaustive classification of chemical components, but focuses on structuring terms according to widely accepted categories. This class is independent of, but may be used in conjunction with, classes that describe ‘analytical methods’ for quantification, ‘physical properties’ or ‘dietary function’. Quantification data may be used and reported in research literature, to inform food composition tables and labelling, or for supply chain quality assurance and control. More specifically, terms within the ‘nutritional component concentration’ class may be used to represent quantification of components described in the ‘dietary nutritional component’ class. Concentration data are intended to be described in conjunction with post-composed metadata concepts, such as represented by the Food Ontology (FoodOn) ‘Food product by organism’, which derives from some food or anatomical entity and a NCBI organismal classification ontology (NCBITaxon) entity. The common vocabulary and relationships defined within CDNO should facilitate description, communication and exchange of material entity-derived nutritional composition datasets typically generated by analytical laboratories. The organisation of the vocabulary is structured to reflect common categories variously used by those involved in crop, livestock or other organismal production, associated R&amp;D and breeding, as well as the food processing and supply sector, and nutritionists, inlcuding compilers and users of food composition databases. The CDNO therefore supports characterisation of genetic diversity and management of biodiversity collections, as well as sharing of knowledge relating to dietary composition between a wider set of researchers, breeders, farmers, processors and other stakeholders. Further development of the functional class should also assist in understanding how interactions between organismal genetic and environmental variation contribute to human diet and health in the farm to fork continuum.","An anatomical and developmental ontology for cephalopods","A structured classification of molecular entities of biological interest focusing on 'small' chemical compounds.","Includes terms for the descriptors commonly used in cheminformatics software applications and the algorithms which generate them.","CHEBI provides a distinct role hierarchy. Chemicals in the structural hierarchy are connected via a 'has role' relation. CHIRO provides links from these roles to useful other classes in other ontologies. This will allow direct connection between chemical structures (small molecules, drugs) and what they do. This could be formalized using 'capable of', in the same way Uberon and the Cell Ontology link structures to processes.","CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).","The Ontology of Coronavirus Infectious Disease (CIDO) is a community-driven open-source biomedical ontology in the area of coronavirus infectious disease. The CIDO is developed to provide standardized human- and computer-interpretable annotation and representation of various coronavirus infectious diseases, including their etiology, transmission, pathogenesis, diagnosis, prevention, and treatment.","An ontology to capture confidence information about annotations.","An ontology of cell types.","CLAO is an ontology of anatomical terms employed in morphological descriptions for the Class Collembola (Arthropoda: Hexapoda).","The Cell Line Ontology (CLO) is a community-based ontology of cell lines. The CLO is developed to unify publicly available cell line entry data from multiple sources to a standardized logically defined format based on consensus design patterns.","Anatomy, development and life cycle stages - planula, polyp, medusa/jellyfish - of the cnidarian hydrozoan species, Clytia hemiphaerica.","Morphological and physiological measurement records generated from clinical and model organism research and health programs.","CMPO is a species neutral ontology for describing general phenotypic observations relating to the whole cell, cellular components, cellular processes and cell populations.","Rice Trait Dictionary in template v 5.0 - IRRI - March 2016 - Based on SES, RD, UPOV variables and on variables used by CIAT, FLAR and the GRISP Phenotyping Network variables","July 2018","Maize Trait Dictionary in template 5 - CIMMYT- September 2016","ICARDA - TDv5 - Sept 2018","Sorghum TDv5 - Oct 2019","Banana Trait Dictionary in template 5 - Bioversity &amp; IITA - April 2019","Draft version","Pearl millet Trait Dictionary in template 5 - ICRISAT/INERA - April 2016","CIP - potato ontology - december 2018","Sweet Potato Trait Dictionary in template v5 - CIP - November 2019","This ontology was built as part of the AKER project. It describes variables used in beet phenotyping (experimental properties and measurement scale) for each institution (INRAE, Geves, ITB) and breeding companies (Florimond Desprez). Curator: Dorothee Charruaud (ADRINORD - URGI) Daphne Verdelet (Florimond Desprez) - First submission in November 2017.","Cassava Trait Dictionary in template 5 - IITA - July 2015, updated in February 2016","CIAT Common bean trait dictionary - version August 2014","Soybean Trait Dictionary in template v5 - IITA - July 2015","Groundnut Trait Dictionary in template v5 - ICRISAT/ISRA/DARS - Sept 2015","Chickpea Trait Dictionary in template v5 - ICRISAT - July 2015","Lentil Trait Dictionary in template v5 - ICARDA - July 2015","Cowpea Trait Dictionary in template v5 - IITA - August 2015","Pigeonpea Trait Dictionary in template v5 - ICRISAT - July 2015","version 2019 - pvs","Brachiaria (forages) ontology TD v5 - Version Oct 2016","oct 2016","March 2017 version ","Brassica Trait Ontology (BRaTO) hosts trait information to describe brassica crop data. Terms are collected from various projects including OREGIN, RIPR (UK) and Rapsodyn (France). BRATO development is conducted by Earlham Institute (UK), Southern Cross University (Australia) and INRA (France).","Oat trait dictionary started by Oat Global (http://oatglobal.org/) and improved by NIAB and PepsiCo","Grape Ontology including OIV and bioversity descriptors. INRA July 2017","This ontology lists all variables used for woody plant observations. Terms are collected from various sources (past and ongoing projects at national and international levels). Curators: Celia Michotey (INRAE) &amp; Ines Chaves (IBET) - Version 2 submitted on Jun 2020 by INRAE.","Cotton ontology from CottonGen database - June 2019","December 2019","Sugar Kelp trait ontology","developed by ICARDA - Dec 2018","version Dec 2019","COB brings together key terms from a wide range of OBO projects to improve interoperability.","The Coleoptera Anatomy Ontology contains terms used for describing the anatomy and phenotype of beetles in biodiversity research.","The COVID-19 Vocabulary (COVoc) is an ontology containing terms related to the research of the COVID-19 pandemic. This includes host organisms, pathogenicity, gene and gene products, barrier gestures, treatments and more.","A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.","Ontology that describes data types and image processing operations in Cryo Electron Microscopy of Single Particles","An anatomical and developmental ontology for ctenophores (Comb Jellies)","The core Ontology of Clinical Trials (CTO) will serve as a structured resource integrating basic terms and concepts in the context of clinical trials. Thereby covering clinicaltrails.gov. CoreCTO will serve as a basic ontology to generate extended versions for specific applications such as annotation of variables in study documents from clinical trials.","An ontology to describe entities related to cardiovascular diseases",null,null,"A structured controlled vocabulary of anatomies of the slime-mold Dictyostelium discoideum.","A structured controlled vocabulary of phenotypes of the slime-mould Dictyostelium discoideum.","DICOM Controlled Terminology","The Potential Drug-drug Interaction and Potential Drug-drug Interaction Evidence Ontology","Drivers of human diseases including environmental, maternal and social exposures.","The Disease Ontology has been developed as a standardized ontology for human disease with the purpose of providing the biomedical community with consistent, reusable and sustainable descriptions of human disease terms, phenotype characteristics and related medical vocabulary disease concepts.","An ontology for the description of Drosophila melanogaster phenotypes.","An ontology to support comparative effectiveness researchers studying claims data.","DUO is an ontology which represent data use conditions.","None","The Evidence &amp; Conclusion Ontology (ECO) describes types of scientific evidence within the biological research domain that arise from laboratory experiments, computational methods, literature curation, or other means.","Ecocore is a community ontology for the concise and controlled description of ecological traits of organisms.","ECTO describes exposures to experimental treatments of plants and model organisms (e.g. exposures to modification of diet, lighting levels, temperature); exposures of humans or any other organisms to stressors through a variety of routes, for purposes of public health, environmental monitoring etc, stimuli, natural and experimental, any kind of environmental condition or change in condition that can be experienced by an organism or population of organisms on earth. The scope is very general and can include for example plant treatment regimens, as well as human clinical exposures (although these may better be handled by a more specialized ontology).","EDAM is a domain ontology of data analysis and data management in bio- and other sciences, and science-based applications. It comprises concepts related to analysis, modelling, optimisation, and data life cycle. Targetting usability by diverse users, the structure of EDAM is relatively simple, divided into 4 main sections: Topic, Operation, Data (incl. Identifier), and Format.","The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org","A structured controlled vocabulary of stage-specific anatomical structures of the developing human.","A structured controlled vocabulary of stage-specific anatomical structures of the mouse (Mus).","An ontology for mouse anatomy covering embryonic development and postnatal stages.","The eNanoMapper project (https://www.enanomapper.net/), NanoCommons project (https://www.nanocommons.eu/) and ACEnano project (http://acenano-project.eu/) are creating a pan-European computational infrastructure for toxicological data management for ENMs, based on semantic web standards and ontologies. This ontology is an application ontology targeting the full domain of nanomaterial safety assessment. It re-uses several other ontologies including the NPO, CHEMINF, ChEBI, and ENVO.","The Ensembl glossary lists the terms, data types and file types that are used in Ensembl and describes how they are used.","ENVO is an ontology which represents knowledge about environments,environmental processes, ecosystems, habitats, and related entities","L'ontologie EOL décrit les conditions d'environnement des élevages d'animaux domestiques. Elle décrit plus particulièrement les modalités de l'alimentation, de l'environnement, de la structure des élevages et des systèmes d'élevage","An ontology of research resources such as instruments. protocols, reagents, animal models and biospecimens.","The VEuPathDB ontology is an application ontology developed to encode our understanding of what data is about in the public resources developed and maintained by the Eukaryotic Pathogen, Host and Vector Genomics Resource (VEuPathDB; https://veupathdb.org). The VEuPathDB ontology was previously named the EuPathDB ontology prior to EuPathDB joining with VectorBase.The ontology was built based on the Ontology of Biomedical Investigations (OBI) with integration of other OBO ontologies such as PATO, OGMS, DO, etc. as needed for coverage. Currently the VEuPath ontology is primarily intended to be used for support of the VEuPathDB sites. Terms with VEuPathDB ontology IDs that are not specific to VEuPathDB will be submitted to OBO Foundry ontologies for subsequent import and replacement of those terms when they are available.","ExO is intended to bridge the gap between exposure science and diverse environmental health disciplines including toxicology, epidemiology, disease surveillance, and epigenetics.","A structured controlled vocabulary for the anatomy of fungi.","A structured controlled vocabulary of sample preparation, visualization and imaging methods used in biomedical research.","An ontology representing the gross anatomy of Drosophila melanogaster.","A miscellaneous ontology of terms used for curation in FlyBase, including the DPO.","An ontology of Drosophila melanogaster developmental stages.","The taxonomy of the family &lt;i&gt;Drosophilidae&lt;/i&gt; (largely after Baechli) and of other taxa referred to in FlyBase.","Food-Drug interactions automatically extracted from scientific literature","An ontology of physico-chemical methods and properties.","Traits and phenotypes of flowering plants occurring in digitized Floras","This is currently a slimmed down version of FMA","FOBI (Food-Biomarker Ontology) is an ontology to represent food intake data and associate it with metabolomic data","A broadly scoped ontology representing entities which bear a “food role”. It encompasses materials in natural ecosystems and agriculture that are consumed by humans and domesticated animals. This includes any generic (unbranded) raw or processed food material found in processing plants, markets, stores or food distribution points. FoodOn also imports nutritional component and dietary pattern terms from other OBO Foundry ontologies to support interoperability in diet and nutrition research","These are the terms that are improted for FOVT to describe vertebrate traits.","A formal ontology of phenotypes observed in fission yeast.","A gazetteer constructed on ontological principles. The countries are actively maintained.","An ontology to represent genomics cohort attributes.","The Genomic Epidemiology Ontology (GenEpiO) covers vocabulary necessary to identify, document and research foodborne pathogens and associated outbreaks.","GENO is an OWL model of genotypes, their more fundamental sequence components, and links to related biological and experimental entities.  At present many parts of the model are exploratory and set to undergo refactoring.  In addition, many classes and properties have GENO URIs but are place holders for classes that will be imported from an external ontology (e.g. SO, ChEBI, OBI, etc).  Furthermore, ongoing work will implement a model of genotype-to-phenotype associations. This will support description of asserted and inferred relationships between a genotypes, phenotypes, and environments, and the evidence/provenance behind these associations. \n\nDocumentation is under development as well, and for now a slidedeck is available at http://www.slideshare.net/mhb120/brush-icbo-2013","An ontology of geographical entities","Gene Expression Ontology","An ontology for glycans based on GlyTouCan, but organized by subsumption.","The Gene Ontology (GO) provides a framework and set of concepts for describing the functions of gene products from all organisms.","GSSO is the Gender, Sex, and Sex Orientation ontology, including terms related to gender identity and expression, sexual and romantic identity and orientation, and sexual and reproductive behavior.","Human ancestry ontology for the NHGRI GWAS Catalog","A structured controlled vocabulary of the anatomy of the Hymenoptera (bees, wasps, and ants)","Application ontology for human cell types, anatomy and development stages for the Human Cell Atlas.","This ontology represents concepts related to homology, as well as other concepts used to describe similarity and non-homology.","The Human Phenotype Ontology (HPO) provides a standardized vocabulary of phenotypic abnormalities and clinical features encountered in human disease.","Life cycle stages for Human","The health Surveillance Ontology (HSO) focuses on \"surveillance system level data\", that is, data outputs from surveillance activities, such as number of samples collected, cases observed, etc. It aims to support One-Health surveillance, covering animal health, public health and food safety surveillance.","An ontology for representing clinical data about hypertension, intended to support classification of patients according to various diagnostic guidelines","The Information Artifact Ontology (IAO) is an ontology of information entities, originally driven by work by the OBI digital entity and realizable information entity branch.","A biological ontology to standardize and integrate Integrative and Conjugative Element (ICE) information and to support computer-assisted reasoning.","The Informed Consent Ontology (ICO) is an ontology for the informed consent and informed consent process in the medical field.","A set of interoperable ontologies that will together provide coverage of the infectious disease domain. IDO core is the upper-level ontology that hosts terms of general relevance across the domain, while extension ontologies host terms to specific to a particular part of the domain.","The COVID-19 Infectious Disease Ontology (IDO-COVID-19) is an extension of the Infectious Disease Ontology (IDO) and the Virus Infectious Disease Ontology (VIDO). IDO-COVID-19 follows OBO Foundry guidelines, employs the Basic Formal Ontology as its starting point, and covers epidemiology, classification, pathogenesis, and treatment of terms used to represent infection by the SARS-CoV-2 virus strain, and the associated COVID-19 disease.","An application ontology to cover all aspects of malaria as well as the intervention attempts to control it.","he Interaction Network Ontology (INO) is an ontology in the domain of interactions and interaction networks. INO represents general and species-neutral types of interactions and interaction networks, and their related elements and relations. INO is a community-driven ontology, aligns with BFO, and is developed by following the OBO Foundry principles.","LABO is an ontology of informational entities formalizing clinical laboratory tests prescriptions and reporting documents.","A vocabulary for cattle, chicken, horse, pig, and sheep breeds.","The Lepidoptera Anatomy Ontology contains terms used for describing the anatomy and phenotype of moths and butterflies in biodiversity research.","A structured controlled vocabulary of the adult anatomy of the mouse (Mus).","The Mathematical Modelling Ontology (MAMO) is a classification of the types of mathematical models used mostly in the life sciences, their variables, relationships and other relevant features.","An ontology to represent medically relevant actions, procedures, therapies, interventions, and recommendations.","Microbial Conditions Ontology is an ontology...","The Mental Functioning Ontology is an overarching ontology for all aspects of mental functioning.","The Mammalian Feeding Muscle Ontology is an antomy ontology for the muscles of the head and neck that participate in feeding, swallowing, and other oral-pharyngeal behaviors.","An ontology of affective phenomena such as emotions, moods, appraisals and subjective feelings.","The Mental Disease Ontology is developed to facilitate representation for all aspects of mental disease. It is an extension of the Ontology for General Medical Science (OGMS) and Mental Functioning Ontology (MF).","A structured controlled vocabulary for the annotation of experiments concerned with protein-protein interactions.","The MIAPA ontology is intended to be an application ontology for the purpose of semantic annotation of phylogenetic data according to the requirements and recommendations of the Minimum Information for A Phylogenetic Analysis (MIAPA) metadata reporting standard. The ontology leverages (imports) primarily from the CDAO (Comparative Data Analysis Ontology), PROV (W3C Provenance Ontology), and SWO (Software Ontology, which includes the EDAM ontologies) ontologies. It adds some assertions of its own, as well as some classes and individuals that may eventually get pushed down into one of the respective source ontologies.\n\nThis ontology is maintained at http://github.com/miapa/miapa, and requests for changes or additions should be filed at the issue tracker there. The discussion list is at miapa-discuss@googlegroups.com. Further resources about MIAPA can be found at the project's main page at http://evoio.org/wiki/MIAPA.","An ontology of prokaryotic phenotypic and metabolic characters","Application ontology for entities related to insecticide resistance in mosquitos","A representation of the variety of methods used to make clinical and phenotype measurements.","Life cycle stages for Mus Musculus","PSI-MOD is an ontology consisting of terms that describe protein chemical modifications","A semi-automatically constructed ontology that merges in multiple disease resources to yield a coherent merged ontology.","MOP is the molecular process ontology. It contains the molecular processes that underlie the name reaction ontology RXNO, for example cyclization, methylation and demethylation.","The Mammalian Phenotype Ontology is being developed by Cynthia L. Smith, Susan M. Bello, Anna, Anagnostopoulos, Carroll W. Goldsmith and Janan T. Eppig, as part of the Mouse Genome Database (MGD) Project, Mouse Genome Informatics (MGI), The Jackson Laboratory, Bar Harbor, ME. This file contains pre-coordinated phenotype terms, definitions and synonyms that can be used to describe mammalian phenotypes. The ontology is represented as a directed acyclic graph (DAG). It organizes phenotype terms into major biological system headers such as nervous system and respiratory system.  This ontology is currently under development. Weekly updates are available at the Mouse Genome Informatics (MGI) ftp site (ftp://ftp.informatics.jax.org/pub/reports/index.html#pheno) as well as the OBO Foundry site (http://obofoundry.org/). Japanese labels of MP were created by the MP Japanese Translation Project (https://github.com/dbcls/MP_Japanse). Questions, comments and suggestions are welcome, and should be directed to pheno@jax.org, Susan.Bello@jax.org or to GitHub tracker (https://github.com/mgijax/mammalian-phenotype-ontology/issues) MGD is funded by NIH/NHGRI grant HG000330.","A structured controlled vocabulary of mutant and transgenic mouse pathology phenotypes","An ontology of minimum information regarding potential drug-drug interaction information.","The MHC Restriction Ontology is an application ontology capturing how Major Histocompatibility Complex (MHC) restriction is defined in experiments, spanning exact protein complexes, individual protein chains, serotypes, haplotypes and mutant molecules, as well as evidence for MHC restrictions.","A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.","an application ontology for supporting description and annotation of mass-spectrometry and nmr-spectroscopy based metabolomics experiments and fluxomics studies.","An ontology of human and animal behaviours and behavioural phenotypes","An ontology representation of the NCBI organismal taxonomy","NCI Thesaurus (NCIt)is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. The NCIt OBO Edition aims to increase integration of the NCIt with OBO Library ontologies. NCIt OBO Edition releases should be considered experimental.","An ontology for non-coding RNA, both of biological origin, and engineered.","Next Generation Biobanking Ontology (NGBO) is an open application ontology representing contextual data about omics digital assets in biobank. The ontology focuses on capturing the information about three main activities: wet bench analysis used to generate omics data, bioinformatics analysis used to analyze and interpret data, and data management.","This artefact is an MSI-approved controlled vocabulary primarily developed under COSMOS EU and PhenoMeNal EU governance. The nmrCV is supporting the nmrML XML format with standardized terms. nmrML is a vendor agnostic open access NMR raw data standard. Its primaly role is analogous to the mzCV for the PSI-approved mzML XML format. It uses BFO2.0 as its Top level. This CV was derived from two predecessors (The NMR CV from the David Wishart Group, developed by Joseph Cruz) and the MSI nmr CV developed by Daniel Schober at the EBI.  This simple taxonomy of terms (no DL semantics used) serves the nuclear magnetic resonance markup language (nmrML) with meaningful descriptors to amend the nmrML xml file with CV terms. Metabolomics scientists are encouraged to use this CV to annotrate their raw and experimental context data, i.e. within nmrML. The approach to have an exchange syntax mixed of an xsd and CV stems from the PSI mzML effort. The reason to branch out from an xsd into a CV is, that in areas where the terminology is likely to change faster than the nmrML xsd could be updated and aligned, an externally and decentrallised maintained CV can accompensate for such dynamics in a more flexible way. A second reason for this set-up is that semantic validity of CV terms used in an nmrML XML instance (allowed CV terms, position/relation to each other, cardinality) can be validated by rule-based proprietary validators:  By means of cardinality specifications and XPath expressions defined in an XML mapping file (an instances of the CvMappingRules.xsd ), one can define what ontology terms are allowed in a specific location of the data model.","NOMEN is a nomenclatural ontology for biological names (not concepts).  It encodes the goverened rules of nomenclature.","The Ontology of Adverse Eventsy (OAE) is a biomedical ontology in the domain of adverse events. OAE aims to standardize adverse event annotation, integrate various adverse event data, and support computer-assisted reasoning.  OAE is a community-based ontology. Its development follows the OBO Foundry principles. Vaccine adverse events have been used as an initial testing use case. OAE also studies adverse events associated with the administration of drug and nutritional products, the operation of surgeries, and the usage of medical devices, etc.","OArCS is an ontology describing the Arthropod ciruclatory system.","A collection of biological attributes (traits) covering all kingdoms of life. Interoperable with VT (vertebrate trait ontology) and TO (plant trait ontology). Extends PATO.","OBCS stands for the Ontology of Biological and Clinical Statistics. OBCS is an ontology in the domain of biological and clinical statistics. It is aligned with the Basic Formal Ontology (BFO) and the Ontology for Biomedical Investigations (OBI). OBCS imports all possible biostatistics terms in OBI and includes many additional biostatistics terms, some of which were proposed and discussed in the OBI face-to-face workshop in Ann Arbor in 2012.","An ontology for representing biomedical investigations, including study designs, the collection and preparation of the targets of investigation, assays, instrumentation and reagents used, as well as the data generated and the types of analysis performed on the data to reach conclusions, and their documentation.","The Ontology for Biobanking (OBIB) is an ontology for the annotation and modeling of the activities, contents, and administration of a biobank. Biobanks are facilities that store specimens, such as bodily fluids and tissues, typically along with specimen annotation and clinical data. OBIB is based on a subset of the Ontology for Biomedical Investigation (OBI), has the Basic Formal Ontology (BFO) as its upper ontology, and is developed following OBO Foundry principles. The first version of OBIB resulted from the merging of two existing biobank-related ontologies, OMIABIS and biobank ontology.",null,"OGG is a biological ontology in the area of genes and genomes. OGG uses the Basic Formal Ontology (BFO) as its upper level ontology. This OGG document contains the genes and genomes of a list of selected organisms, including human, two viruses (HIV and influenza virus), and bacteria (B. melitensis strain 16M, E. coli strain K-12 substrain MG1655, M. tuberculosis strain H37Rv, and P. aeruginosa strain PAO1). More OGG information for other organisms (e.g., mouse, zebrafish, fruit fly, yeast, etc.) may be found in other OGG subsets.","OGI formalized the genomic element by defining an upper class 'genetic interval'.\n\nThe definition of 'genetic interval' is \"the spatial continuous physical entity which contains ordered genomic sets(DNA, RNA, Allele, Marker,etc.) between and including two points (Nucleic Acid Base Residue) on a chromosome or RNA molecule which must have a liner primary sequence sturcture.\" \n\nRelated paper:\n\n1. Yu Lin, Norihiro Sakamoto (2009) “Genome, Gene, Interval and Ontology” Interdisciplinary Ontology Vol.2 - Proceedings of the Second Interdisciplinary Meeting, Tokyo, Feb. 28th- Mar. 1st, 2009. Page(s):25-34 (http://cdb-riken.academia.edu/LinYu/Papers/142399/Genome_Gene_Interval_and_Ontology)\nYu Lin, Hiroshi Tarui, Peter Simons (2009) “From Ontology for Genetic Interval(OGI) to Sequence Assembly – Ontology apply to next generation sequencing” Proceeding of the Semantic Web Applications and Tools for Life Science Workshop, Amsterdam, Nov.20th, 2009. (http://ceur-ws.org/Vol-559/Poster2.pdf)\nYu Lin, Peter Simons (2010) “DNA sequence from below: A Nominalist Approach” Interdisciplinary Ontology Vol.3 - Proceedings of the Second Interdisciplinary Meeting, Tokyo, Feb. 28th- Mar. 1st, 2010. (http://philpapers.org/rec/LINDSF)\n\n\n","The Ontology for General Medical Science (OGMS) is an ontology of entities involved in a clinical encounter. OGMS includes very general terms that are used across medical disciplines, including: 'disease', 'disorder', 'disease course', 'diagnosis', 'patient', and 'healthcare provider'. OGMS uses the Basic Formal Ontology (BFO) as an upper-level ontology. The scope of OGMS is restricted to humans, but many terms can be applied to a variety of organisms. OGMS provides a formal theory of disease that can be further elaborated by specific disease ontologies. This theory is implemented using OWL-DL and OBO Relation Ontology relations and is available in OWL and OBO formats.\n\nOGMS is based on the papers Toward an Ontological Treatment of Disease and Diagnosis and On Carcinomas and Other Pathological Entities. The ontology attempts to address some of the issues raised at the Workshop on Ontology of Diseases (Dallas, TX) and the Signs, Symptoms, and Findings Workshop(Milan, Italy). OGMS was formerly called the clinical phenotype ontology. Terms from OGMS hang from the Basic Formal Ontology.","An application ontology to represent genetic susceptibility to a specific disease, adverse event, or a pathological process.","The Oral Health and Disease Ontology is used for representing the diagnosis and treatment of dental maladies.","OHMI is a biomedical ontology that represents the entities and relations in the domain of host-microbiome interactions.","OHPI is a biomedical ontology in the area of host-pathogen interactions. OHPI is developed by following the OBO Foundry Principles (e.g., openness and collaboration).","Life cycle stages for Medaka","The OM ontology provides classes, instances, and properties that represent the different concepts used for defining and using measures and units. It includes, for instance, common units such as the SI units meter and kilogram, but also units from other systems of units such as the mile or nautical mile. For many application areas it includes more specific units and quantities, such as the unit of the Hubble constant: km/s/Mpc, or the quantity vaselife.  OM defines the complete set of concepts in the domain as distinguished in the textual standards. As a result the ontology can answer a wider range of competency questions than the existing approaches do. The following application areas are supported by OM:  Geometry; Mechanics; Thermodynamics; Electromagnetism; Fluid mechanics; Chemical physics; Photometry; Radiometry and Radiobiology; Nuclear physics; Astronomy and Astrophysics; Cosmology; Earth science; Meteorology; Material science; Microbiology; Economics; Information technology; Typography; Shipping; Food engineering; Post-harvest; technology; Dynamics of texture and taste; Packaging","An ontological version of MIABIS (Minimum Information About BIobank data Sharing)","Ontology to establish data exchange standards and common data elements in the microRNA (miR) domain","An ontology specifies terms that are used to annotate ontology terms for all OBO ontologies. The ontology was developed as part of Information Artifact Ontology (IAO).","An ontology of phenotypes covering microbes","The Ontology for Modeling and Representation of Social Entities (OMRSE) is an OBO Foundry ontology that represents the various entities that arise from human social interactions, such as social acts, social roles, social groups, and organizations.","An ontology to standardize research output of nutritional epidemiologic studies.","An ontology for description of concepts in the nutritional studies domain.","The Ontology for Avida (OntoAvida) project aims to develop an integrated vocabulary for the description of the most widely used computational approach for performing experimental evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment).\n\nThe lack of a clearly defined vocabulary makes biologists feel reluctant to embrace the field of digital evolution. This unique ontology has the potential to change this picture overnight.\n\nOntoAvida was initially developed by https://fortunalab.org, the computational biology lab at the Doñana Biological Station (a research institute of the Spanish National Research Council based at Seville, Spain). Contributors to OntoAvida are expected to include members of the Digital Evolution Laboratory (https://devolab.org/) at Michigan State University (USA).\n\nMore information can be found at https://obofoundry.org/ontology/ontoavida.html","The Obstetric and Neonatal Ontology is a structured controlled vocabulary to provide a representation of the data from electronic health records (EHRs) involved in the care of the pregnant woman, and of her baby.","The Ontology of Organizational Structures of Trauma centers and Trauma systems (OOSTT) is a representation of the components of trauma centers and trauma systems coded in Web Ontology Language (OWL2).","The Ontology for Parasite Lifecycle (OPL) models the life cycle stage details of various parasites, including Trypanosoma sp., Leishmania major, and Plasmodium sp., etc. In addition to life cycle stages, the ontology also models necessary contextual details, such as host information, vector information, and anatomical location. OPL is based on the Basic Formal Ontology (BFO) and follows the rules set by the OBO Foundry consortium.","OPMI is a biomedical ontology in the area of precision medicine and its related investigations. It is community-driven and developed by following the OBO Foundry ontology development principles.","An application ontology designed to annotate next-generation sequencing experiments performed on RNA.","The Orphanet Rare Disease ontology (ORDO) is jointly developed by Orphanet and the EBI to provide a structured vocabulary for rare diseases capturing relationships between diseases, genes and other relevant features which will form a useful resource for the computational analysis of rare diseases. It derived from the Orphanet database (www.orpha.net ) , a multilingual database dedicated to rare diseases populated from literature and validated by international experts. It integrates a nosology (classification of rare diseases), relationships (gene-disease relations, epiemological data) and connections with other terminologies (MeSH, UMLS, MedDRA),databases (OMIM, UniProtKB, HGNC, ensembl, Reactome, IUPHAR, Geantlas) or classifications (ICD10).","The need of a common ontology for describing orthology information in biological research communities has led to the creation of the Orthology Ontology (ORTH). ORTH ontology is designed to describe sequence homology data available in multiple orthology databases on the Web (e.g.: OMA, OrthoDB, HieranoiDB, and etc.). By sequence homology data, we mostly mean gene region, gene and protein centric orthology, paralogy, and xenology information. Depending on the database, the homology information is structured in different ways. ORTH ontology accommodates these disparate data structures namely Hierarchical Orthologous Group (HOG), cluster of homologous sequences and homologous-pairwise relations between sequences. In addition to the specific ORTH terms, this specification includes terms of the imported ontologies (e.g. Semanticscience Integrated Ontology, SIO) which are pertinents to represent the information from various orthology databases in a homogeneous way.","OVAE is a biomedical ontology in the area of vaccine adverse events. OVAE is an extension of the community-based Ontology of Adverse Events (OAE).",null,"An ontology of phenotypic qualities (properties, attributes or characteristics).","Cell types that are provisionally defined by experimental techniques such as single cell transcriptomics rather than a straightforward &amp; coherent set of properties.","The Population and Community Ontology (PCO) describes material entities, qualities, and processes related to collections of interacting organisms such as populations and communities. It is taxon neutral, and can be used for any species, including humans. The classes in the PCO are useful for describing evolutionary processes, organismal interactions, and ecological experiments. Practical applications of the PCO include community health care, plant pathology, behavioral studies, sociology, and ecology.","An ontology to describe entities related to prescription of drugs","Life cycle stages for Platynereis dumerilii","A structured, controlled vocabulary which describes the treatments, growing conditions, and/or study types used in plant biology experiments.","Ontology of species-neutral phenotypes observed in pathogen-host interactions.","PLANA, the PLANarian Anatomy Ontology, encompasses the anatomy of developmental stages and adult biotypes of Schmidtea mediterranea.","Planarian Phenotype Ontology is an ontology of phenotypes observed in the planarian Schmidtea mediterranea.","The Plant Ontology is a structured vocabulary and database resource that links plant anatomy, morphology and growth and development to plant genomics data.","An ontology describing the anatomical structures and characteristics of Porifera (sponges)","An ontology for describing the phenology of individual plants and populations of plants, and for integrating plant phenological data across sources and scales.","An ontological representation of protein-related entities","The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.","ProbOnto, is an ontology-based knowledge base of probability distributions, featuring more than eighty uni- and multivariate distributions with their defining functions, characteristics, relationships and reparameterisation formulas.","PROCO (PROcess Chemistry Ontology) is a formal ontology that aims to standardly represent entities and relations among entities in the domain of process chemistry.",null,"Performance Summary Display Ontology (PSDO) is an application ontology about motivating information in performance summaries and the visual and textual entities that are used to communicate about performance. Motivating information includes performance comparisons and changes that motivate improvement or sustainment, such as improvement towards a goal, loss of high performer status, or the presence of a performance gap. Visual and textual entities include charts, tables, and graphs that display performance information. PSDO's domain focus is healthcare organizations that use clinical quality dashboards and feedback interventions for healthcare professionals and teams. Performance information is commonly about the quality of care and health outcomes that have been derived from clinical data using performance measures (aka metrics, process indicators, quality measures, etc). PSDO uses Basic Formal Ontology as its upper level ontology. This work is supported by the NIH, National Library of Medicine (1K01LM012528-01, 1R01LM013894-01).\n\nLandis-Lewis Z, Stansbury C, Rincón J, Gross C. Performance Summary Display Ontology: Feedback intervention content, delivery, and interpreted information. International Conference on Biomedical Ontology 2022 (ICBO 2022). https://icbo-conference.github.io/icbo2022/papers/ICBO-2022_paper_2172.pdf","The Plant Stress Ontology describes biotic and abiotic stresses that a plant may encounter.","A controlled vocabulary for annotating gene products to pathways.","RBO is an ontology for the effects of radiation on biota in terrestrial and space environments.",null,"The REPRODUCE-ME ontology is an extension of the PROV-O and the P-Plan ontology to describe a complete path of a scientific experiment. It expresses the REPRODUCE-ME Data Model using the OWL2 Web Ontology Language (OWL2). It provides a set of classes and properties to represent a scientific experiment including its computational and non-computational steps to track the provenance of results. It describes a complete path of a scientific experiment considering the use-case of biological imaging and microscopy experiments, computational experiments, including Jupyter notebooks and scripts. It describes an experiment and its data, agents, activities, plans, steps, variables, instruments, materials, and settings required for its reproducibility.","Regulation of Transcription","An ontology of physico-chemical processes, i.e. physico-chemical changes occurring in course of time.","Regulation of Gene Expression","Controlled vocabulary pertaining to RNA function and based on RNA sequences, secondary and three-dimensional structures.","Relationship types shared across multiple ontologies","Ontology of rat strains","RXNO is the name reaction ontology. It contains more than 500 classes representing organic reactions such as the Diels–Alder cyclization.","Terms commonly used in Systems Biology, and in particular in computational modeling.","An ontology for the standardization of terminology and integration of knowledge about Sickle Cell Disease.",null,"An OBO-compliant ontology representing the entities referenced by the SDGs, their targets, and indicators.","An ontology for representing the provenance of scientific claims and the evidence that supports them.","ShareLoc defines terms to annotate data sets from single molecule localization microscopy, including but not limited to: imaging technique, biological structures or molecules of interest, cell types, experimental condition, labeling method, fixation protocol, etc.","Social Behavior in insects","The Semanticscience Integrated Ontology (SIO) provides a simple, integrated ontology of types and relations for rich description of objects, processes and their attributes.",null,"SwissLipids is a curated resource that provides information about known lipids, including lipid structure, metabolism, interactions, and subcellular and tissue localization. Information is curated from peer-reviewed literature and referenced using established ontologies, and provided with full provenance and evidence codes for curated assertions.","SNOMED CT or SNOMED Clinical Terms is a systematically organized computer processable collection of medical terms providing codes, terms, synonyms and definitions used in clinical documentation and reporting.","A structured controlled vocabulary for sequence annotation, for the exchange of annotation data and for the description of sequence objects in databases.","An ontology for spider comparative biology including anatomical parts (e.g. leg, claw), behavior (e.g. courtship, combing) and products (i.g. silk, web, borrow).","The FAIRsharing Subject Ontology (SRAO) is an application ontology for the categorization of research disciplines across all research domains, from the humanities to the natural sciences. It utilizes multiple external vocabularies.","STATO is the statistical methods ontology. It contains concepts and properties related to statistical methods, probability distributions and other concepts related to statistical analysis, including relationships to study designs and plots.","The Software Ontology (SWO) is a resource for describing software tools, their types, tasks, versions, provenance and associated data. It contains detailed information on licensing and formats as well as software applications themselves, mainly (but not limited) to the bioinformatics community.","The Symptom Ontology has been developed as a standardized ontology for symptoms of human diseases.","A terminology for the skills necessary to make data FAIR and to keep it FAIR.","The anatomy of the Tick, &lt;i&gt;Families: Ixodidae, Argassidae&lt;/i&gt;","Multispecies fish anatomy ontology. Originally seeded from ZFA, but intended to cover terms relevant to other taxa","A vocabulary of taxonomic ranks (species, family, phylum, etc)","A structured controlled vocabulary of the anatomy of mosquitoes.","A controlled vocabulary to describe phenotypic traits in plants.","The Pathogen Transmission Ontology describes the tranmission methods of human disease pathogens describing how a pathogen is transmitted from one host, reservoir, or source to another host. The pathogen transmission may occur either directly or indirectly and may involve animate vectors or inanimate vehicles.","An ontology covering the taxonomy of teleosts (bony fish)","Elucidating the mechanism of toxicity is crucial in drug safety evaluations. TOXic Process Ontology (TXPO) systematizes a wide variety of terms involving toxicity courses and processes. The first version of TXPO focuses on liver toxicity.\n\nThe TXPO contains an is-a hierarchy that is organized into three layers: the top layer contains general terms, mostly derived from the Basic Formal Ontology. The intermediate layer contains biomedical terms in OBO foundry from UBERON, Cell Ontology, NCBI Taxon, ChEBI, Gene Ontology, PATO, OGG, INOH, HINO, NCIT, DOID and Relational ontology (RO). The lower layer contains toxicological terms.\n\nIn applied work, we have developed a prototype of TOXPILOT, a TOXic Process InterpretabLe knOwledge sysTem. TOXPILOT provides visualization maps of the toxic course, which facilitates capturing the comprehensive picture for understanding toxicity mechanisms. A prototype of TOXPILOT is available: https://toxpilot.nibiohn.go.jp","Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.","Unimod is a community supported, comprehensive database of protein modifications for mass spectrometry applications. That is, accurate and verifiable values, derived from elemental compositions, for the mass differences introduced by all types of natural and artificial modifications. Other important information includes any mass change, (neutral loss), that occurs during MS/MS analysis, an d site specificity, (which residues are susceptible to modification and any constraints on the position of the modification within the protein or peptide).","Metrical units for use in conjunction with PATO","A manually curated resource for the representation and annotation of metabolic pathways","The uPheno ontology integrates multiple phenotype ontologies into a unified cross-species phenotype ontology.","Variation Ontology, VariO, is an ontology for standardized, systematic description of effects, consequences and mechanisms of variations.","Vertebrate Breed Ontology is an ontology created to serve as a single computable resource for vertebrate breed names.","The Virus Infectious Disease Ontology (IDO Virus) is an extension of the Infectious Disease Ontology (IDO). IDO Virus follows OBO Foundry guidelines, employs the Basic Formal Ontology as its starting point, and covers epidemiology, classification, pathogenesis, and treatment of terms used by Virologists, i.e. virus, prion, satellite, viroid, etc.","The Vaccine Ontology (VO) is a community-based biomedical ontology in the domain of vaccine and vaccination. VO aims to standardize vaccine types and annotations, integrate various vaccine data, and support computer-assisted reasoning. The VO supports basic vaccine R&amp;D and clincal vaccine usage. VO is being developed as a community-based ontology with support and collaborations from the vaccine and bio-ontology communities.","Vertebrate skeletal anatomy ontology.","The Vertebrate Trait Ontology is a controlled vocabulary for the description of traits (measurable or observable characteristics) pertaining to the morphology, physiology, or development of vertebrate organisms.","Comprehensive hierarchy of extinct and extant vertebrate taxa.","Ontology about the gross anatomy of the C. elegans","Ontology about the development and life stages of the C. elegans","Ontology about C. elegans and other nematode phenotypes","XAO represents the anatomy and development of the African frogs Xenopus laevis and tropicalis.","Conditions under which physiological and morphological measurements are made both in the clinic and in studies involving humans or model organisms.","A structured controlled vocabulary for cross-linking reagents used with proteomics mass spectrometry.","A structured controlled vocabulary for cross-linking reagents used with proteomics mass spectrometry.","XPO represents anatomical, cellular, and gene function phenotypes occurring throughout the development of the African frogs Xenopus laevis and tropicalis.","Ontology of Zebrafish Experimental Conditions","ZFA description.","Developmental stages of the Zebrafish","The Zebrafish Phenotype Ontology formally defines all phenotypes of the Zebrafish model organism."],["http://purl.obolibrary.org/obo/ado.owl","http://purl.obolibrary.org/obo/aeo.owl",null,null,"http://purl.obolibrary.org/obo/agro.owl","http://purl.obolibrary.org/obo/aism.owl","http://purl.obolibrary.org/obo/amphx.owl","http://purl.obolibrary.org/obo/apo.owl","http://purl.obolibrary.org/obo/apollo_sv.owl","http://purl.obolibrary.org/obo/aro.owl",null,null,null,"http://humanbehaviourchange.org/ontology/bcio.owl","http://purl.obolibrary.org/obo/bco.owl","http://purl.obolibrary.org/obo/bfo.owl",null,"http://purl.obolibrary.org/obo/kisao.owl","file:///nfs/production/parkinso/spot/ols4/prod/local_ontologies/teddy-inferred-fixed.owl","http://purl.obolibrary.org/obo/bspo.owl","http://purl.obolibrary.org/obo/bto.owl","https://champ.stuchalk.domains.unf.edu/images/ontology/cao.owl","http://purl.obolibrary.org/obo/caro.owl","https://purl.org/ccf/2.0/ccf.owl","http://www.bio.ntnu.no/ontology/CCO/cco.owl","http://purl.obolibrary.org/obo/cdao.owl","http://purl.obolibrary.org/obo/cdno.owl","http://purl.obolibrary.org/obo/ceph.owl","http://purl.obolibrary.org/obo/chebi.owl","http://purl.obolibrary.org/obo/cheminf.owl","http://purl.obolibrary.org/obo/chiro.owl","http://purl.obolibrary.org/obo/chmo.owl","http://purl.obolibrary.org/obo/cido.owl","http://purl.obolibrary.org/obo/cio.owl","http://purl.obolibrary.org/obo/cl.owl","http://purl.obolibrary.org/obo/clao.owl","http://purl.obolibrary.org/obo/clo.owl","http://purl.obolibrary.org/obo/clyh.owl","http://purl.obolibrary.org/obo/cmo.owl","http://www.ebi.ac.uk/cmpo/cmpo.owl",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"http://purl.obolibrary.org/obo/cob.owl","http://purl.obolibrary.org/obo/colao.owl","https://github.com/EBISPOT/covoc/releases/download/current/covoc.owl","http://purl.obolibrary.org/obo/cro.owl","http://scipion.i2pc.es/ontology/cryoem.owl","http://purl.obolibrary.org/obo/cteno.owl",null,"http://purl.obolibrary.org/obo/cvdo.owl",null,null,"http://purl.obolibrary.org/obo/ddanat.owl","http://purl.obolibrary.org/obo/ddpheno.owl","ftp://medical.nema.org/MEDICAL/Dicom/Resources/Ontology/DCM/dcm.owl","http://purl.obolibrary.org/obo/dideo.owl","http://purl.obolibrary.org/obo/disdriv.owl","http://purl.obolibrary.org/obo/doid.owl","http://purl.obolibrary.org/obo/dpo.owl","http://purl.obolibrary.org/obo/dron.owl","http://purl.obolibrary.org/obo/duo.owl","http://purl.obolibrary.org/obo/ecao.owl","http://purl.obolibrary.org/obo/eco.owl","http://purl.obolibrary.org/obo/ecocore.owl","http://purl.obolibrary.org/obo/ecto.owl","https://raw.githubusercontent.com/edamontology/edamontology/master/releases/EDAM.owl","http://www.ebi.ac.uk/efo/efo.owl","http://purl.obolibrary.org/obo/ehdaa2.owl",null,"http://purl.obolibrary.org/obo/emapa.owl","http://enanomapper.github.io/ontologies/enanomapper.owl","https://raw.githubusercontent.com/Ensembl/ensembl-glossary/master/ensembl-glossary.owl","http://purl.obolibrary.org/obo/envo.owl",null,null,"http://purl.obolibrary.org/obo/eupath.owl","http://purl.obolibrary.org/obo/exo.owl","http://purl.obolibrary.org/obo/fao.owl","http://purl.obolibrary.org/obo/fbbi.owl","http://purl.obolibrary.org/obo/fbbt.owl","http://purl.obolibrary.org/obo/fbcv.owl","http://purl.obolibrary.org/obo/fbdv.owl",null,"http://purl.obolibrary.org/obo/fideo.owl","http://purl.obolibrary.org/obo/fix.owl","http://purl.obolibrary.org/obo/flopo.owl","http://purl.obolibrary.org/obo/fma.owl","http://purl.obolibrary.org/obo/fobi.owl","http://purl.obolibrary.org/obo/foodon.owl","http://purl.obolibrary.org/obo/fovt.owl","http://purl.obolibrary.org/obo/fypo.owl","http://purl.obolibrary.org/obo/gaz.owl","http://purl.obolibrary.org/obo/gecko.owl","http://purl.obolibrary.org/obo/genepio.owl","http://purl.obolibrary.org/obo/geno.owl","http://purl.obolibrary.org/obo/geo.owl",null,"http://purl.obolibrary.org/obo/gno.owl","http://purl.obolibrary.org/obo/go/extensions/go-plus.owl","http://purl.obolibrary.org/obo/gsso.owl","http://purl.obolibrary.org/obo/hancestro.owl","http://purl.obolibrary.org/obo/hao.owl","https://raw.githubusercontent.com/ebi-ait/ontology/master/hcao.owl","http://purl.obolibrary.org/obo/hom.owl","http://purl.obolibrary.org/obo/hp/hp-international.owl","http://purl.obolibrary.org/obo/hsapdv.owl","http://purl.obolibrary.org/obo/hso.owl","http://purl.obolibrary.org/obo/htn.owl","http://purl.obolibrary.org/obo/iao.owl","http://purl.obolibrary.org/obo/iceo.owl","http://purl.obolibrary.org/obo/ico.owl","http://purl.obolibrary.org/obo/ido.owl",null,"http://purl.obolibrary.org/obo/idomal.owl","http://purl.obolibrary.org/obo/ino.owl","http://purl.obolibrary.org/obo/labo.owl",null,"http://purl.obolibrary.org/obo/lepao.owl","http://purl.obolibrary.org/obo/ma.owl",null,"http://purl.obolibrary.org/obo/maxo.owl","http://purl.obolibrary.org/obo/mco.owl","http://purl.obolibrary.org/obo/mf.owl","http://purl.obolibrary.org/obo/mfmo.owl","http://purl.obolibrary.org/obo/mfoem.owl","http://purl.obolibrary.org/obo/mfomd.owl","http://purl.obolibrary.org/obo/mi.owl","http://purl.obolibrary.org/obo/miapa.owl","http://purl.obolibrary.org/obo/micro.owl","http://purl.obolibrary.org/obo/miro.owl","http://purl.obolibrary.org/obo/mmo.owl","http://purl.obolibrary.org/obo/mmusdv.owl","http://purl.obolibrary.org/obo/mod.owl","http://purl.obolibrary.org/obo/mondo.owl","http://purl.obolibrary.org/obo/mop.owl","http://purl.obolibrary.org/obo/mp/mp-international.owl","http://purl.obolibrary.org/obo/mpath.owl","http://purl.obolibrary.org/obo/mpio.owl","http://purl.obolibrary.org/obo/mro.owl","http://purl.obolibrary.org/obo/ms.owl","https://raw.githubusercontent.com/MSI-Metabolomics-Standards-Initiative/MSIO/master/releases/latest_release/MSIO-merged-reasoned.owl","http://purl.obolibrary.org/obo/nbo.owl","http://purl.obolibrary.org/obo/ncbitaxon.owl","http://purl.obolibrary.org/obo/ncit.owl","http://purl.obolibrary.org/obo/ncro.owl","http://purl.obolibrary.org/obo/ngbo.owl","http://nmrml.org/cv/stable/nmrCV.owl","http://purl.obolibrary.org/obo/nomen.owl","http://purl.obolibrary.org/obo/oae.owl","http://purl.obolibrary.org/obo/oarcs.owl","http://purl.obolibrary.org/obo/oba.owl","http://purl.obolibrary.org/obo/obcs.owl","http://purl.obolibrary.org/obo/obi.owl","http://purl.obolibrary.org/obo/obib.owl",null,"http://purl.obolibrary.org/obo/ogg.owl",null,"http://purl.obolibrary.org/obo/ogms.owl","http://purl.obolibrary.org/obo/ogsf.owl","http://purl.obolibrary.org/obo/ohd.owl","http://purl.obolibrary.org/obo/ohmi.owl","http://purl.obolibrary.org/obo/ohpi.owl","http://purl.obolibrary.org/obo/olatdv.owl",null,null,"http://purl.obolibrary.org/obo/omit.owl","http://purl.obolibrary.org/obo/omo.owl","http://purl.obolibrary.org/obo/omp.owl","http://purl.obolibrary.org/obo/omrse.owl","http://purl.obolibrary.org/obo/one.owl","http://purl.obolibrary.org/obo/ons.owl","http://purl.obolibrary.org/obo/ontoavida.owl","http://purl.obolibrary.org/obo/ontoneo.owl","http://purl.obolibrary.org/obo/oostt.owl","http://purl.obolibrary.org/obo/opl.owl","http://purl.obolibrary.org/obo/opmi.owl","http://purl.obolibrary.org/obo/ornaseq.owl","http://www.orphadata.org/data/ORDO/ordo_orphanet.owl",null,"http://purl.obolibrary.org/obo/ovae.owl",null,"http://purl.obolibrary.org/obo/pato.owl","http://purl.obolibrary.org/obo/pcl.owl","http://purl.obolibrary.org/obo/pco.owl","http://purl.obolibrary.org/obo/pdro.owl","http://purl.obolibrary.org/obo/pdumdv.owl","http://purl.obolibrary.org/obo/peco.owl","http://purl.obolibrary.org/obo/phipo.owl","http://purl.obolibrary.org/obo/plana.owl","http://purl.obolibrary.org/obo/planp.owl","http://purl.obolibrary.org/obo/po.owl","http://purl.obolibrary.org/obo/poro.owl","http://purl.obolibrary.org/obo/ppo.owl","http://purl.obolibrary.org/obo/pr.owl","https://raw.githubusercontent.com/PRIDE-Utilities/pride-ontology/master/pride_cv.owl",null,"http://purl.obolibrary.org/obo/proco.owl",null,"http://purl.obolibrary.org/obo/psdo.owl","http://purl.obolibrary.org/obo/pso.owl","http://purl.obolibrary.org/obo/pw.owl","http://purl.obolibrary.org/obo/rbo.owl",null,null,null,"http://purl.obolibrary.org/obo/rex.owl",null,null,"http://purl.obolibrary.org/obo/ro.owl","http://purl.obolibrary.org/obo/rs.owl","http://purl.obolibrary.org/obo/rxno.owl","http://purl.obolibrary.org/obo/sbo.owl","http://purl.obolibrary.org/obo/scdo.owl",null,"http://purl.unep.org/sdg/sdgio.owl","http://purl.obolibrary.org/obo/sepio.owl","https://raw.githubusercontent.com/imodpasteur/ShareLoc.XYZ/ontology/shareloc.owl","http://purl.obolibrary.org/obo/sibo.owl","https://raw.githubusercontent.com/micheldumontier/semanticscience/master/ontology/sio/release/sio-release.owl",null,null,"file:///nfs/production/parkinso/spot/ols4/prod/local_ontologies/snomed-inferred.owl","http://purl.obolibrary.org/obo/so.owl","http://purl.obolibrary.org/obo/spd.owl","https://github.com/FAIRsharing/subject-ontology/raw/master/SRAO.owl","http://purl.obolibrary.org/obo/stato.owl","http://purl.obolibrary.org/obo/swo.owl","http://purl.obolibrary.org/obo/symp.owl","http://purl.obolibrary.org/obo/t4fs.owl","http://purl.obolibrary.org/obo/tads.owl",null,"http://purl.obolibrary.org/obo/taxrank.owl","http://purl.obolibrary.org/obo/tgma.owl","http://purl.obolibrary.org/obo/to.owl","http://purl.obolibrary.org/obo/trans.owl","http://purl.obolibrary.org/obo/tto.owl","http://purl.obolibrary.org/obo/txpo.owl","http://purl.obolibrary.org/obo/uberon.owl","https://raw.githubusercontent.com/PRIDE-Utilities/pride-ontology/master/unimod.owl","http://purl.obolibrary.org/obo/uo.owl","http://purl.obolibrary.org/obo/upa.owl","https://github.com/obophenotype/upheno-dev/releases/latest/download/upheno.owl",null,"http://purl.obolibrary.org/obo/vbo.owl",null,"http://purl.obolibrary.org/obo/vo.owl",null,"http://purl.obolibrary.org/obo/vt.owl","http://purl.obolibrary.org/obo/vto.owl","http://purl.obolibrary.org/obo/wbbt.owl","http://purl.obolibrary.org/obo/wbls.owl","http://purl.obolibrary.org/obo/wbphenotype.owl","http://purl.obolibrary.org/obo/xao.owl","http://purl.obolibrary.org/obo/xco.owl",null,"http://purl.obolibrary.org/obo/xlmod.owl","http://purl.obolibrary.org/obo/xpo.owl","http://purl.obolibrary.org/obo/zeco.owl","http://purl.obolibrary.org/obo/zfa.owl","http://purl.obolibrary.org/obo/zfs.owl","http://purl.obolibrary.org/obo/zp.owl"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>description<\/th>\n      <th>download_owl<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"description","targets":1},{"name":"download_owl","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script><p>We can use the URLs given in this table to explore ontologies of interest. For example, the AEO (anatomical entity ontology) extends CARO (the common anatomy reference ontology). What sorts of terms are regarded as extensions?</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aeo</span> <span class="op">=</span> <span class="fu"><a href="../reference/owl2cache.html">owl2cache</a></span><span class="op">(</span>url<span class="op">=</span><span class="st">"http://purl.obolibrary.org/obo/aeo.owl"</span><span class="op">)</span> <span class="co"># localize OWL</span></span></code></pre></div>
<pre><code><span><span class="co">## resource BFC1057 already in cache from http://purl.obolibrary.org/obo/aeo.owl</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aeoinr</span> <span class="op">=</span> <span class="fu"><a href="../reference/setup_entities2.html">setup_entities2</a></span><span class="op">(</span><span class="va">aeo</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span> <span class="co"># zero-length angle</span></span>
<span><span class="fu"><a href="../reference/onto_plot2.html">onto_plot2</a></span><span class="op">(</span><span class="va">aeoinr</span>, <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"AEO"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">aeoinr</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>, value<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,<span class="fl">12</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="ontoProc_files/figure-html/dodemo11-1.png" width="700"></p>
<p>So CARO is already using some of the extensions.</p>
<p>Use a search facility in owlready2 to check the UBERON ontology (check the table above for references) for terms involving the substring ‘vein’:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ub</span> <span class="op">=</span> <span class="fu"><a href="../reference/owl2cache.html">owl2cache</a></span><span class="op">(</span>url<span class="op">=</span><span class="st">"http://purl.obolibrary.org/obo/uberon.owl"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## resource BFC1060 already in cache from http://purl.obolibrary.org/obo/uberon.owl</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">allv</span> <span class="op">=</span> <span class="fu"><a href="../reference/search_labels.html">search_labels</a></span><span class="op">(</span><span class="va">ub</span>, <span class="st">"*vein*"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">allv</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 362</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">allv</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                             UBERON_0001638 </span></span>
<span><span class="co">##                                     "vein" </span></span>
<span><span class="co">##                                 CL_0002543 </span></span>
<span><span class="co">##                    "vein endothelial cell" </span></span>
<span><span class="co">##                                 CL_0002588 </span></span>
<span><span class="co">## "smooth muscle cell of the umbilical vein" </span></span>
<span><span class="co">##                             UBERON_0002066 </span></span>
<span><span class="co">##                           "umbilical vein" </span></span>
<span><span class="co">##                                 CL_0002618 </span></span>
<span><span class="co">##       "endothelial cell of umbilical vein" </span></span>
<span><span class="co">##                                 CL_0009094 </span></span>
<span><span class="co">##  "endothelial cell of hepatic portal vein"</span></span></code></pre>
<p>It is interesting to note that owlready2 includes the ability to provide <a href="https://owlready2.readthedocs.io/en/v0.47/annotations.html#full-text-search-fts" class="external-link">relevance measurement</a> for search results using the <a href="https://en.wikipedia.org/wiki/Okapi_BM25" class="external-link">BM25 index</a>. We need to add some code to capitalize on this in ontoProc.</p>
</div>
<div class="section level3">
<h3 id="legacy-approaches">Legacy approaches<a class="anchor" aria-label="anchor" href="#legacy-approaches"></a>
</h3>
<p>The following table describes the most up-to-date resources available with <code>getOnto</code>. <!-- use inst/scripts/desc.R to generate all but the 'purpose' which is added by hand --></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">packDesc2023</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">packDesc2023</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">7</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">name</th>
<th align="right">nclass</th>
<th align="right">nprop</th>
<th align="right">nroots</th>
<th align="left">datav</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">caro</td>
<td align="right">751</td>
<td align="right">57</td>
<td align="right">30</td>
<td align="left">caro/releases/2022-02-18/caro-full.owl</td>
</tr>
<tr class="even">
<td align="left">cellLineOnto</td>
<td align="right">43596</td>
<td align="right">204</td>
<td align="right">110</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">cellOnto</td>
<td align="right">17371</td>
<td align="right">229</td>
<td align="right">102</td>
<td align="left">releases/2023-02-15</td>
</tr>
<tr class="even">
<td align="left">cellosaurus</td>
<td align="right">144570</td>
<td align="right">NA</td>
<td align="right">144571</td>
<td align="left">44.0</td>
</tr>
<tr class="odd">
<td align="left">chebi_full</td>
<td align="right">163735</td>
<td align="right">34</td>
<td align="right">13</td>
<td align="left">218</td>
</tr>
<tr class="even">
<td align="left">chebi_lite</td>
<td align="right">163735</td>
<td align="right">32</td>
<td align="right">13</td>
<td align="left">218</td>
</tr>
<tr class="odd">
<td align="left">diseaseOnto</td>
<td align="right">13680</td>
<td align="right">26</td>
<td align="right">3</td>
<td align="left">doid/releases/2023-01-30/doid-non-classified.obo</td>
</tr>
<tr class="even">
<td align="left">efoOnto</td>
<td align="right">40514</td>
<td align="right">93</td>
<td align="right">60</td>
<td align="left"><a href="http://www.ebi.ac.uk/efo/releases/v3.51.0/efo.owl" class="external-link uri">http://www.ebi.ac.uk/efo/releases/v3.51.0/efo.owl</a></td>
</tr>
<tr class="odd">
<td align="left">goOnto</td>
<td align="right">47427</td>
<td align="right">41</td>
<td align="right">10</td>
<td align="left">releases/2023-01-01</td>
</tr>
<tr class="even">
<td align="left">hcaoOnto</td>
<td align="right">22821</td>
<td align="right">259</td>
<td align="right">161</td>
<td align="left">releases/2022-12-16/hcao.owl</td>
</tr>
<tr class="odd">
<td align="left">mondo</td>
<td align="right">44707</td>
<td align="right">249</td>
<td align="right">116</td>
<td align="left">releases/2022-08-01</td>
</tr>
<tr class="even">
<td align="left">patoOnto</td>
<td align="right">2813</td>
<td align="right">45</td>
<td align="right">16</td>
<td align="left">releases/2022-12-15/pato.obo</td>
</tr>
<tr class="odd">
<td align="left">PROnto</td>
<td align="right">334873</td>
<td align="right">48</td>
<td align="right">49</td>
<td align="left">68.0</td>
</tr>
<tr class="even">
<td align="left">uberon</td>
<td align="right">25836</td>
<td align="right">245</td>
<td align="right">138</td>
<td align="left">releases/2023-02-14</td>
</tr>
<tr class="odd">
<td align="left">Pronto</td>
<td align="right">315957</td>
<td align="right">6</td>
<td align="right">53</td>
<td align="left">57.0</td>
</tr>
</tbody>
</table>
<p>Other resources are listed in <code>packDesc202x</code> and <code>packDesc2019</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h2>
<div class="section level3">
<h3 id="conceptual-overview-of-ontology-with-cell-types">Conceptual overview of ontology with cell types<a class="anchor" aria-label="anchor" href="#conceptual-overview-of-ontology-with-cell-types"></a>
</h3>
<p><strong>Definitions, semantics.</strong> For concreteness, we provide some definitions and examples. We use <code>ontology</code> to denote the systematic organization of terminology used in a conceptual domain. The <code>Cell Ontology</code> is a graphical data structure with carefully annotated terms as nodes and conventionally defined semantic relationships among terms serving as edges. As an example, <code>lung ciliated cell</code> has URI . This URI includes a fixed-length identifier <code>CL_1000271</code> with unambiguous interpretation wherever it is encountered. There is a chain of relationships from <code>lung ciliated cell</code> up through <code>ciliated cell</code>, then <code>native cell</code>, then <code>cell</code>, each possessing its own URI and related interpretive metadata. The relationship connecting the more precise to the less precise term in this chain is denoted <code>SubclassOf</code>. <code>Ciliated cell</code> is equivalent to a <code>native cell</code> that <code>has plasma membrane part</code> <code>cilium</code>. Semantic characteristics of terms and relationships are used to infer relationships among terms that may not have relations directly specified in available ontologies.</p>
<p><strong>Barriers to broad adoption.</strong> Given the wealth of material available in biological ontologies, it is somewhat surprising that formal annotation is so seldom used in practice. Barriers to more common use of ontology in data annotation include: (i) Non-existence of exact matching between intended term and terms available in ontologies of interest. (ii) The practical problem of decoding ontology identifiers. A GO tag or CL tag is excellent for programming, but it is clumsy to co-locate with the tag the associated natural language term or phrase. (iii) Likelihood of disagreement of suitability of terms for conditions observed at the boundaries of knowledge. To help cope with the first of these problems, Bioconductor’s <code>ontologyProc</code> package includes a function <code>liberalMap</code> which will search an ontology for terms lexically close to some target term or phrase. The second problem can be addressed with more elaborate data structures for variable annotation and programming in R, and the third problem will diminish in importance as the value of ontology adoption becomes manifest in more applications.</p>
<p><strong>Class vs. instance.</strong> It is important to distinguish the practice of designing and maintaining ontologies from the use of ontological class terms to annotate instances of the concepts. The combination of an ontology and a set of annotated instances is called a knowledge base. To illustrate some of the salient distinctions here, consider the cell line called A549, which is established from a human lung adenocarcinoma sample. There is no mention of A549 in the Cell Ontology. However, A549 is present in the EBI Experimental Factor Ontology as a subclass of the “Homo sapiens cell line” class. Presumably this is because A549 is a class of cells that are widely used experimentally, and this cell line constitutes a concept deserving of mapping in the universe of experimental factors. In the universe of concepts related to cell structure and function <em>per se</em>, A549 is an individual that can be characterized through possession of or lack of properties enumerated in Cell Ontology, but it is not deserving of inclusion in that ontology.</p>
</div>
<div class="section level3">
<h3 id="illustration-in-a-single-cell-rna-seq-dataset">Illustration in a single-cell RNA-seq dataset<a class="anchor" aria-label="anchor" href="#illustration-in-a-single-cell-rna-seq-dataset"></a>
</h3>
<p>The 10X Genomics corporation has distributed a dataset on results of sequencing 10000 PBMC from a healthy donor . Subsets of the data are used in tutorials for the Seurat analytical suite (<span class="citation">Butler et al. (2018)</span>).</p>
<div class="section level4">
<h4 id="labeling-pbmc-in-the-seurat-tutorial">Labeling PBMC in the Seurat tutorial<a class="anchor" aria-label="anchor" href="#labeling-pbmc-in-the-seurat-tutorial"></a>
</h4>
<p>One result of the tutorial analysis of the 3000 cell subset is a table of cell types and expression-based markers of cell identity. The first three columns of the table below are from concluding material in the Seurat tutorial; the remaining columns are created by “manual” matching between the Seurat terms and terms found in Cell Ontology.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="va">stab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/seur3kTab.html">seur3kTab</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">grp</th>
<th align="left">markers</th>
<th align="left">seurTutType</th>
<th align="left">formal</th>
<th align="left">tag</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">0</td>
<td align="left">IL7R</td>
<td align="left">CD4 T cells</td>
<td align="left">CD4-positive helper T cell</td>
<td align="left">CL:0000492</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">CD14, LYZ</td>
<td align="left">CD14+ Monocytes</td>
<td align="left">CD14-positive monocyte</td>
<td align="left">CL:0001054</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="left">MS4A1</td>
<td align="left">B cells</td>
<td align="left">B cell</td>
<td align="left">CL:0000236</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="left">CD8A</td>
<td align="left">CD8 T cells</td>
<td align="left">CD8-positive, alpha-beta T cell</td>
<td align="left">CL:0000625</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="left">FCGR3A, MS4A7</td>
<td align="left">FCGR3A+ Monocytes</td>
<td align="left">monocyte</td>
<td align="left">CL:0000576</td>
</tr>
<tr class="even">
<td align="right">5</td>
<td align="left">GNLY, NKG7</td>
<td align="left">NK cells</td>
<td align="left">natural killer cell</td>
<td align="left">CL:0000623</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="left">FCER1A, CST3</td>
<td align="left">Dendritic Cells</td>
<td align="left">dendritic cell</td>
<td align="left">CL:0000451</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="left">PPBP</td>
<td align="left">Megakaryocytes</td>
<td align="left">megakaryocyte</td>
<td align="left">CL:0000556</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="relationships-asserted-in-the-cell-ontology">Relationships asserted in the Cell Ontology<a class="anchor" aria-label="anchor" href="#relationships-asserted-in-the-cell-ontology"></a>
</h4>
<p>Given the informally selected tags in the table above, we can sketch the Cell Ontology graph connecting the associated cell types. The ontoProc package adds functionality to ontologyPlot with <code>make_graphNEL_from_ontology_plot</code>. This allows use of all Rgraphviz and igraph visualization facilities for graphs derived from ontology structures.</p>
<p>Here we display the PBMC cell sets reported in the Seurat tutorial.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn/ontoProc">ontoProc</a></span><span class="op">)</span></span>
<span><span class="va">cl</span> <span class="op">=</span> <span class="fu"><a href="../reference/getOnto.html">getOnto</a></span><span class="op">(</span><span class="st">"cellOnto"</span>, <span class="st">"2023"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/onto_plot2.html">onto_plot2</a></span><span class="op">(</span><span class="va">cl</span>, <span class="va">stab</span><span class="op">$</span><span class="va">tag</span><span class="op">)</span></span></code></pre></div>
<p><img src="ontoProc_files/figure-html/lklklk-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="molecular-features-asserted-in-the-cell-ontology">Molecular features asserted in the Cell Ontology<a class="anchor" aria-label="anchor" href="#molecular-features-asserted-in-the-cell-ontology"></a>
</h4>
<p>The <code>CLfeats</code> function traces relationships and properties from a given Cell Ontology class. Briefly, each class can assert that it is the <code>intersection_of</code> other classes, and <code>has_part</code>, <code>lacks_part</code>, <code>has_plasma_membrane_part</code>, <code>lacks_plasma_membrane_part</code> can be asserted as relationships holding between cell type instances and cell components. The components are often cross-referenced to Protein Ontology or Gene Ontology. When the Protein Ontology component has a synonym for which an HGNC symbol is provided, that symbol is retrieved by <code>CLfeats</code>. Here we obtain the listing for a mature CD1a-positive dermal dendritic cell.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="fu"><a href="../reference/CLfeats.html">CLfeats</a></span><span class="op">(</span><span class="va">cl</span>, <span class="st">"CL:0002531"</span>, pr<span class="op">=</span><span class="va">pr</span>, go<span class="op">=</span><span class="va">go</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="10%">
<col width="12%">
<col width="8%">
<col width="47%">
<col width="6%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">tag</th>
<th align="left">prtag</th>
<th align="left">cond</th>
<th align="left">entity</th>
<th align="left">SYMBOL</th>
<th align="left">name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">CL:0000451</td>
<td align="left">PR:000001002</td>
<td align="left">lacksPMP</td>
<td align="left">CD19 molecule</td>
<td align="left">CD19</td>
<td align="left">dendritic cell</td>
</tr>
<tr class="even">
<td align="left">CL:0000451</td>
<td align="left">PR:000001003</td>
<td align="left">lacksPMP</td>
<td align="left">CD34 molecule</td>
<td align="left">CD34</td>
<td align="left">dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">CL:0000451</td>
<td align="left">PR:000001020</td>
<td align="left">lacksPMP</td>
<td align="left">CD3 epsilon</td>
<td align="left">CD3E</td>
<td align="left">dendritic cell</td>
</tr>
<tr class="even">
<td align="left">CL:0000451</td>
<td align="left">PR:000001024</td>
<td align="left">lacksPMP</td>
<td align="left">neural cell adhesion molecule 1</td>
<td align="left">NCAM1</td>
<td align="left">dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">CL:0000451</td>
<td align="left">PR:000001289</td>
<td align="left">lacksPMP</td>
<td align="left">membrane-spanning 4-domains subfamily A member 1</td>
<td align="left">MS4A1</td>
<td align="left">dendritic cell</td>
</tr>
<tr class="even">
<td align="left">CL:0000451</td>
<td align="left"><a href="GO:0042613" class="uri">GO:0042613</a></td>
<td align="left">hasPart</td>
<td align="left">MHC class II protein complex</td>
<td align="left">NA</td>
<td align="left">dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">CL:0000000</td>
<td align="left"><a href="GO:0005634" class="uri">GO:0005634</a></td>
<td align="left">hasPart</td>
<td align="left">nucleus</td>
<td align="left">NA</td>
<td align="left">cell</td>
</tr>
<tr class="even">
<td align="left">CL:0000000</td>
<td align="left"><a href="GO:0005634" class="uri">GO:0005634</a></td>
<td align="left">hasPart</td>
<td align="left">nucleus</td>
<td align="left">NA</td>
<td align="left">cell</td>
</tr>
</tbody>
</table>
<p>The <code>ctmarks</code> function starts a shiny app that generates tables of this sort for selected cell types.</p>
<div class="figure">
<img src="ctmarks.png" alt=""><p class="caption">ctmarks snapshot</p>
</div>
</div>
<div class="section level4">
<h4 id="mapping-from-gene-presencerole-to-cell-type">Mapping from gene ‘presence/role’ to cell type<a class="anchor" aria-label="anchor" href="#mapping-from-gene-presencerole-to-cell-type"></a>
</h4>
<p>The <code>sym2CellOnto</code> function helps find mention of given gene symbols in properties or parts of cell types.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="va">sdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="../reference/sym2CellOnto.html">sym2CellOnto</a></span><span class="op">(</span><span class="st">"ITGAM"</span>, <span class="va">cl</span>, <span class="va">pr</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="6%">
<col width="10%">
<col width="12%">
<col width="69%">
</colgroup>
<thead><tr class="header">
<th align="left">sym</th>
<th align="left">cond</th>
<th align="left">cl</th>
<th align="left">type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000037</td>
<td align="left">hematopoietic stem cell</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000547</td>
<td align="left">proerythroblast</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000553</td>
<td align="left">megakaryocyte progenitor cell</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000558</td>
<td align="left">reticulocyte</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000611</td>
<td align="left">eosinophil progenitor cell</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000613</td>
<td align="left">basophil progenitor cell</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000765</td>
<td align="left">erythroblast</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000831</td>
<td align="left">mast cell progenitor</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000836</td>
<td align="left">promyelocyte</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000837</td>
<td align="left">hematopoietic multipotent progenitor cell</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000872</td>
<td align="left">splenic marginal zone macrophage</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000941</td>
<td align="left">thymic conventional dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000942</td>
<td align="left">thymic plasmacytoid dendritic cell</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000989</td>
<td align="left">CD11c-low plasmacytoid dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0000998</td>
<td align="left">CD8_alpha-negative CD11b-negative dendritic cell</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0001000</td>
<td align="left">CD8_alpha-positive CD11b-negative dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0001029</td>
<td align="left">common dendritic progenitor</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0001060</td>
<td align="left">hematopoietic oligopotent progenitor cell, lineage-negative</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0001066</td>
<td align="left">erythroid progenitor cell, mammalian</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0002010</td>
<td align="left">pre-conventional dendritic cell</td>
</tr>
<tr class="odd">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0002089</td>
<td align="left">group 2 innate lymphoid cell, mouse</td>
</tr>
<tr class="even">
<td align="left">ITGAM</td>
<td align="left">lacksPMP</td>
<td align="left">CL:0002679</td>
<td align="left">natural helper lymphocyte</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sdf</span><span class="op">$</span><span class="va">cond</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## lacksPMP </span></span>
<span><span class="co">##       22</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="../reference/sym2CellOnto.html">sym2CellOnto</a></span><span class="op">(</span><span class="st">"FOXP3"</span>, <span class="va">cl</span>, <span class="va">pr</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="9%">
<col width="13%">
<col width="69%">
</colgroup>
<thead><tr class="header">
<th align="left">sym</th>
<th align="left">cond</th>
<th align="left">cl</th>
<th align="left">type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">FOXP3</td>
<td align="left">hasPart</td>
<td align="left">CL:0000902</td>
<td align="left">induced T-regulatory cell</td>
</tr>
<tr class="even">
<td align="left">FOXP3</td>
<td align="left">hasPart</td>
<td align="left">CL:0000903</td>
<td align="left">natural T-regulatory cell</td>
</tr>
<tr class="odd">
<td align="left">FOXP3</td>
<td align="left">hasPart</td>
<td align="left">CL:0000919</td>
<td align="left">CD8-positive, CD25-positive, alpha-beta regulatory T cell</td>
</tr>
<tr class="even">
<td align="left">FOXP3</td>
<td align="left">hasPart</td>
<td align="left">CL:0000920</td>
<td align="left">CD8-positive, CD28-negative, alpha-beta regulatory T cell</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="adding-terms-to-ontology_index-structures-to-extend-cell-ontology">Adding terms to ontology_index structures to ‘extend’ Cell Ontology<a class="anchor" aria-label="anchor" href="#adding-terms-to-ontology_index-structures-to-extend-cell-ontology"></a>
</h3>
<p>The task of extending an ontology is partly bureaucratic in nature and depends on a collection of endorsements and updates to centralized information structures. In order to permit experimentation with interfaces and new content that may be quite speculative, we include an approach to combining new ontology ‘terms’ of structure similar to those endorsed in Cell Ontology, to ontologyIndex-based <code>ontology_index</code> instances.</p>
<div class="section level4">
<h4 id="use-case-a-set-of-cell-types-defined-by-diagonal-expression">Use case: a set of cell types defined by “diagonal expression”<a class="anchor" aria-label="anchor" href="#use-case-a-set-of-cell-types-defined-by-diagonal-expression"></a>
</h4>
<p>For a demonstration, we consider the discussion in <span class="citation">Bakken et al. (2017)</span>, of a ‘diagonal’ expression pattern defining a group of novel cell types. A set of genes is identified and cells are distinguised by expressing exactly one gene from the set.</p>
<div class="figure">
<img src="usecaseCyclic.png" alt=""><p class="caption">Diagonal expression pattern.</p>
</div>
<p>The necessary information is collected in a vector. The vector is the set of genes, the name of element i is the tag to be associated with the type of cell that expresses gene i and does not express any other gene in the set.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sigels</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CL:X01"</span><span class="op">=</span><span class="st">"GRIK3"</span>, <span class="st">"CL:X02"</span><span class="op">=</span><span class="st">"NTNG1"</span>, <span class="st">"CL:X03"</span><span class="op">=</span><span class="st">"BAGE2"</span>,</span>
<span>             <span class="st">"CL:X04"</span><span class="op">=</span><span class="st">"MC4R"</span>, <span class="st">"CL:X05"</span><span class="op">=</span><span class="st">"PAX6"</span>, <span class="st">"CL:X06"</span><span class="op">=</span><span class="st">"TSPAN12"</span>, </span>
<span>             <span class="st">"CL:X07"</span><span class="op">=</span><span class="st">"hSHISA8"</span>, <span class="st">"CL:X08"</span><span class="op">=</span><span class="st">"SNCG"</span>, <span class="st">"CL:X09"</span><span class="op">=</span><span class="st">"ARHGEF28"</span>, </span>
<span>             <span class="st">"CL:X10"</span><span class="op">=</span><span class="st">"EGF"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="a-data-frame-defining-the-cell-types-and-their-properties">A data.frame defining the cell types and their properties<a class="anchor" aria-label="anchor" href="#a-data-frame-defining-the-cell-types-and-their-properties"></a>
</h4>
<p>The <code>cyclicSigset</code> function produces a data.frame instance connecting cell types with the genes expressed or unexpressed.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cs</span> <span class="op">=</span> <span class="fu"><a href="../reference/cyclicSigset.html">cyclicSigset</a></span><span class="op">(</span><span class="va">sigels</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 100   3</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">9</span><span class="op">:</span><span class="fl">13</span><span class="op">)</span>,<span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##        gene   type     cond</span></span>
<span><span class="co">## 1  ARHGEF28 CL:X09   hasExp</span></span>
<span><span class="co">## 2     GRIK3 CL:X09 lacksExp</span></span>
<span><span class="co">## 3     NTNG1 CL:X09 lacksExp</span></span>
<span><span class="co">## 4     BAGE2 CL:X09 lacksExp</span></span>
<span><span class="co">## 5      MC4R CL:X09 lacksExp</span></span>
<span><span class="co">## 9      SNCG CL:X09 lacksExp</span></span>
<span><span class="co">## 10      EGF CL:X09 lacksExp</span></span>
<span><span class="co">## 11    BAGE2 CL:X03   hasExp</span></span>
<span><span class="co">## 12    GRIK3 CL:X03 lacksExp</span></span>
<span><span class="co">## 13    NTNG1 CL:X03 lacksExp</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cs</span><span class="op">$</span><span class="va">cond</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##   hasExp lacksExp </span></span>
<span><span class="co">##       10       90</span></span></code></pre>
<p>It is expected that a tabular layout like this will suffice to handle general situations of cell type definition.</p>
</div>
<div class="section level4">
<h4 id="translating-the-data-frame-elements-to-obo-term-instances">Translating the data.frame elements to OBO Term instances<a class="anchor" aria-label="anchor" href="#translating-the-data-frame-elements-to-obo-term-instances"></a>
</h4>
<p>The most complicated aspect of novel OBO term construction is the proper specifications of relationships with existing ontology components. A prolog that is mostly shared by all terms is generated programmatically for the diagonal pattern task.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="va">makeIntnProlog</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">id</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span></span>
<span> <span class="co"># make type-specific prologs as key-value pairs</span></span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>  </span>
<span>       <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"id: %s"</span>, <span class="va">id</span><span class="op">)</span>,</span>
<span>       <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"name: %s-expressing cortical layer 1 interneuron, human"</span>, <span class="va">...</span><span class="op">)</span>,</span>
<span>       <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"def: '%s-expressing cortical layer 1 interneuron, human described via RNA-seq observations' [PMID 29322913]"</span>, <span class="va">...</span><span class="op">)</span>,</span>
<span>       <span class="st">"is_a: CL:0000099 ! interneuron"</span>,</span>
<span>       <span class="st">"intersection_of: CL:0000099 ! interneuron"</span><span class="op">)</span></span>
<span> <span class="op">}</span></span></code></pre></div>
<p>The <code>ldfToTerms</code> API uses this to create a set of strings that can be parsed as a term.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pmap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hasExp"</span><span class="op">=</span><span class="st">"has_expression_of"</span>, lacksExp<span class="op">=</span><span class="st">"lacks_expression_of"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">tms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ldfToTerms.html">ldfToTerms</a></span><span class="op">(</span><span class="va">cs</span>, <span class="va">pmap</span>, <span class="va">sigels</span>, <span class="va">makeIntnProlog</span><span class="op">)</span><span class="op">)</span>, <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "[Term]"                                                                                                        </span></span>
<span><span class="co">##  [2] "id: CL:X01"                                                                                                    </span></span>
<span><span class="co">##  [3] "name: GRIK3-expressing cortical layer 1 interneuron, human"                                                    </span></span>
<span><span class="co">##  [4] "def: 'GRIK3-expressing cortical layer 1 interneuron, human described via RNA-seq observations' [PMID 29322913]"</span></span>
<span><span class="co">##  [5] "is_a: CL:0000099 ! interneuron"                                                                                </span></span>
<span><span class="co">##  [6] "intersection_of: CL:0000099 ! interneuron"                                                                     </span></span>
<span><span class="co">##  [7] "has_expression_of: PR:000008242 ! GRIK3"                                                                       </span></span>
<span><span class="co">##  [8] "lacks_expression_of: PR:000011467 ! NTNG1"                                                                     </span></span>
<span><span class="co">##  [9] "lacks_expression_of: PR:000004625 ! BAGE2"                                                                     </span></span>
<span><span class="co">## [10] "lacks_expression_of: PR:000001237 ! MC4R"                                                                      </span></span>
<span><span class="co">## [11] "lacks_expression_of: PR:000012318 ! PAX6"                                                                      </span></span>
<span><span class="co">## [12] "lacks_expression_of: PR:000016738 ! TSPAN12"                                                                   </span></span>
<span><span class="co">## [13] "lacks_expression_of: PR:B8ZZ34 ! hSHISA8"                                                                      </span></span>
<span><span class="co">## [14] "lacks_expression_of: PR:000015325 ! SNCG"                                                                      </span></span>
<span><span class="co">## [15] "lacks_expression_of: PR:000013942 ! ARHGEF28"                                                                  </span></span>
<span><span class="co">## [16] "lacks_expression_of: PR:000006928 ! EGF"                                                                       </span></span>
<span><span class="co">## [17] "[Term]"                                                                                                        </span></span>
<span><span class="co">## [18] "id: CL:X02"                                                                                                    </span></span>
<span><span class="co">## [19] "name: NTNG1-expressing cortical layer 1 interneuron, human"                                                    </span></span>
<span><span class="co">## [20] "def: 'NTNG1-expressing cortical layer 1 interneuron, human described via RNA-seq observations' [PMID 29322913]"</span></span></code></pre>
<p>The content in tms can then be appended to the content of the Cell Ontology cl.obo as text for import with <code><a href="https://rdrr.io/pkg/ontologyIndex/man/get_ontology.html" class="external-link">ontologyIndex::get_OBO</a></code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="subsetting-singler-resources-using-ontological-mapping">Subsetting SingleR resources using ontological mapping<a class="anchor" aria-label="anchor" href="#subsetting-singler-resources-using-ontological-mapping"></a>
</h3>
<div class="section level4">
<h4 id="a-data-frame-mapping-from-informal-to-formal-terms">A data.frame mapping from informal to formal terms<a class="anchor" aria-label="anchor" href="#a-data-frame-mapping-from-informal-to-formal-terms"></a>
</h4>
<p>Aaron Lun has produced a mapping from informal terms used in the Human Primary Cell Atlas to Cell Ontology tags. We provisionally include a copy of this mapping in ontoProc:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hpca_map</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/hpca.csv"</span>, package<span class="op">=</span><span class="st">"ontoProc"</span><span class="op">)</span>, strings<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hpca_map</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                          uncontrolled controlled</span></span>
<span><span class="co">## 1        DC:monocyte-derived:immature CL:0000840</span></span>
<span><span class="co">## 2      DC:monocyte-derived:Galectin-1 CL:0000451</span></span>
<span><span class="co">## 3             DC:monocyte-derived:LPS CL:0000451</span></span>
<span><span class="co">## 4                 DC:monocyte-derived CL:0000451</span></span>
<span><span class="co">## 5 Smooth_muscle_cells:bronchial:vit_D CL:0002598</span></span>
<span><span class="co">## 6       Smooth_muscle_cells:bronchial CL:0002598</span></span></code></pre>
<p>We will rename columns of this map for convenience of our <code>bind_formal_tags</code> method.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hpca_map</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"informal"</span>, <span class="st">"formal"</span><span class="op">)</span>  <span class="co"># obligatory for now</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="binding-formal-tags-to-the-hpca-data">Binding formal tags to the HPCA data<a class="anchor" aria-label="anchor" href="#binding-formal-tags-to-the-hpca-data"></a>
</h4>
<p>I am turning this code off for now because there is no standard approach to getting the mapping from the SummarizedExperment yet. When SingleR merges the ‘standardized’ branch, this will come back.</p>
<p>Let’s retrieve the HPCA data from SingleR:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/celldex" class="external-link">celldex</a></span><span class="op">)</span></span>
<span><span class="va">hpca_sce</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/celldex/man/HumanPrimaryCellAtlasData.html" class="external-link">HumanPrimaryCellAtlasData</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Now bind the formal tags:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hpca_sce</span> <span class="op">=</span> <span class="fu"><a href="../reference/bind_formal_tags.html">bind_formal_tags</a></span><span class="op">(</span><span class="va">hpca_sce</span>, <span class="st">"label.fine"</span>, <span class="va">hpca_map</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">hpca_sce</span><span class="op">$</span><span class="va">label.ont</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 66</span></span></code></pre>
<p>We don’t check for failed mappings:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">hpca_sce</span><span class="op">$</span><span class="va">label.ont</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">hpca_sce</span><span class="op">)</span><span class="op">[</span><span class="va">xx</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 2 rows and 3 columns</span></span>
<span><span class="co">##            label.main             label.fine   label.ont</span></span>
<span><span class="co">##           &lt;character&gt;            &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## GSM154081     T_cells T_cell:CCR10+CLA+1,2..          NA</span></span>
<span><span class="co">## GSM154084     T_cells T_cell:CCR10-CLA+1,2..          NA</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hpca_sce</span><span class="op">$</span><span class="va">label.ont</span> <span class="op">==</span> <span class="st">""</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="co"># iPS and BM</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 50</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="subsetting-using-the-class-hierarchy-of-cell-ontology">Subsetting using the class hierarchy of Cell Ontology<a class="anchor" aria-label="anchor" href="#subsetting-using-the-class-hierarchy-of-cell-ontology"></a>
</h4>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_onto</span> <span class="op">=</span> <span class="fu">ontoProc</span><span class="fu">::</span><span class="fu"><a href="../reference/getOnto.html">getOnto</a></span><span class="op">(</span><span class="st">"cellOnto"</span>, <span class="st">"2023"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hpca_mono</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_descendants.html">subset_descendants</a></span><span class="op">(</span> <span class="va">hpca_sce</span>, <span class="va">cell_onto</span>, <span class="st">"^monocyte$"</span> <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">hpca_mono</span><span class="op">$</span><span class="va">label.fine</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##                          Monocyte             Monocyte:anti-FcgRIIB </span></span>
<span><span class="co">##                                27                                 2 </span></span>
<span><span class="co">##                    Monocyte:CD14+                    Monocyte:CD16- </span></span>
<span><span class="co">##                                 3                                 7 </span></span>
<span><span class="co">##                    Monocyte:CD16+                    Monocyte:CXCL4 </span></span>
<span><span class="co">##                                 6                                 2 </span></span>
<span><span class="co">##   Monocyte:F._tularensis_novicida           Monocyte:leukotriene_D4 </span></span>
<span><span class="co">##                                 6                                 4 </span></span>
<span><span class="co">##                     Monocyte:MCSF Monocyte:S._typhimurium_flagellin </span></span>
<span><span class="co">##                                 2                                 1</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">hpca_mono</span><span class="op">$</span><span class="va">label.ont</span><span class="op">)</span> <span class="co"># not much diversity</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## CL:0000576 CL:0001054 </span></span>
<span><span class="co">##         57          3</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hpca_tcell</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_descendants.html">subset_descendants</a></span><span class="op">(</span> <span class="va">hpca_sce</span>, <span class="va">cell_onto</span>, <span class="st">"^T cell$"</span> <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">hpca_tcell</span><span class="op">$</span><span class="va">label.fine</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##                    T_cell:CD4+     T_cell:CD4+_central_memory </span></span>
<span><span class="co">##                             12                              5 </span></span>
<span><span class="co">##    T_cell:CD4+_effector_memory              T_cell:CD4+_Naive </span></span>
<span><span class="co">##                              4                              6 </span></span>
<span><span class="co">##                    T_cell:CD8+     T_cell:CD8+_Central_memory </span></span>
<span><span class="co">##                             16                              3 </span></span>
<span><span class="co">##    T_cell:CD8+_effector_memory T_cell:CD8+_effector_memory_RA </span></span>
<span><span class="co">##                              4                              4 </span></span>
<span><span class="co">##              T_cell:CD8+_naive                T_cell:effector </span></span>
<span><span class="co">##                              4                              4 </span></span>
<span><span class="co">##             T_cell:gamma-delta              T_cell:Treg:Naive </span></span>
<span><span class="co">##                              2                              2</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">hpca_tcell</span><span class="op">$</span><span class="va">label.ont</span><span class="op">)</span> <span class="co"># </span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## CL:0000624 CL:0000625 CL:0000798 CL:0000895 CL:0000900 CL:0000904 CL:0000905 </span></span>
<span><span class="co">##         12         16          2          6          4          5          4 </span></span>
<span><span class="co">## CL:0000907 CL:0000911 CL:0000913 CL:0001062 CL:0002677 </span></span>
<span><span class="co">##          3          4          4          4          2</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">uu</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">hpca_tcell</span><span class="op">$</span><span class="va">label.ont</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/onto_plot2.html">onto_plot2</a></span><span class="op">(</span><span class="va">cell_onto</span>, <span class="va">uu</span><span class="op">)</span></span></code></pre></div>
<p><img src="ontoProc_files/figure-html/dosub-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="disease-concept-relationships">Disease concept relationships<a class="anchor" aria-label="anchor" href="#disease-concept-relationships"></a>
</h2>
<p>The Experimental Factor Ontology is available and integrates information among diverse ontologies. Here we check on terms likely related to asthma.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ef</span> <span class="op">=</span> <span class="fu"><a href="../reference/getOnto.html">getOnto</a></span><span class="op">(</span><span class="st">"efoOnto"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alla</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"sthma"</span>, <span class="va">ef</span><span class="op">$</span><span class="va">name</span>, value<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="va">aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"obso"</span>, <span class="va">alla</span>, invert<span class="op">=</span><span class="cn">TRUE</span>, value<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/onto_plot2.html">onto_plot2</a></span><span class="op">(</span><span class="va">ef</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">aa</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="ontoProc_files/figure-html/lkefo-1.png" width="700"></p>
<p>However, the Human Disease Ontology seems more developed in terms of defining asthma subtypes. We have not integrated that ontology into ontoProc yet, but it can be retrieved conveniently as follows:</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hdo_2022_09</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_ontology.html" class="external-link">get_OBO</a></span><span class="op">(</span></span>
<span>  <span class="st">"https://github.com/DiseaseOntology/HumanDiseaseOntology/raw/main/src/ontology/HumanDO.obo"</span>, </span>
<span>  extract_tags <span class="op">=</span> <span class="st">"everything"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>With this resource, we can see finer-grained handling of asthma subtyping:</p>
<p><img src="hdoasth.png"></p>
</div>
<div class="section level2">
<h2 id="related-tools">Related tools<a class="anchor" aria-label="anchor" href="#related-tools"></a>
</h2>
<p>Inference on the identities of cells assayed in a single cell transcriptomics experiment can be performed using the Bioconductor <code>celaref</code> package. This package includes a number of reference data resources providing whole-transcriptome profiles of cells of known and unknown type. An approach to systematically structuring data on cell-type signatures, and conducting inference on cell types in new experiments, is provided in the <a href="https://github.com/kevinrue/Hancock" class="external-link">Hancock package</a>, under development.</p>
<p>A CRAN package that is very useful for R programming with ontologies is <code>ontologyIndex</code> <span class="citation">Westbury et al. (2015)</span>. This provides easily used functions for parsing ontologies in the OBO format and for performing basic queries on text fields and list structures.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Aevermann2018" class="csl-entry">
Aevermann, Brian D., Mark Novotny, Trygve Bakken, Jeremy A. Miller, Alexander D. Diehl, David Osumi-Sutherland, Roger S. Lasken, Ed S. Lein, and Richard H. Scheuermann. 2018. <span>“<span class="nocase">Cell type discovery using single-cell transcriptomics: Implications for ontological representation</span>.”</span> <em>Human Molecular Genetics</em> 27 (R1): R40–47. <a href="https://doi.org/10.1093/hmg/ddy100" class="external-link">https://doi.org/10.1093/hmg/ddy100</a>.
</div>
<div id="ref-Bakken2017" class="csl-entry">
Bakken, Trygve, Lindsay Cowell, Brian D. Aevermann, Mark Novotny, Rebecca Hodge, Jeremy A. Miller, Alexandra Lee, et al. 2017. <span>“<span class="nocase">Cell type discovery and representation in the era of high-content single cell phenotyping</span>.”</span> <em>BMC Bioinformatics</em> 18 (Suppl 17). <a href="https://doi.org/10.1186/s12859-017-1977-1" class="external-link">https://doi.org/10.1186/s12859-017-1977-1</a>.
</div>
<div id="ref-butler" class="csl-entry">
Butler, Andrew, Paul Hoffman, Peter Smibert, Efthymia Papalexi, and Rahul Satija. 2018. <span>“Integrating Single-Cell Transcriptomic Data Across Different Conditions, Technologies, and Species.”</span> <em>Nature Biotechnology</em>. <a href="https://doi.org/10.1038/nbt.4096" class="external-link">https://doi.org/10.1038/nbt.4096</a>.
</div>
<div id="ref-Westbury2015" class="csl-entry">
Westbury, Sarah K., Ernest Turro, Daniel Greene, Claire Lentaigne, Anne M. Kelly, Tadbir K. Bariana, Ilenia Simeoni, et al. 2015. <span>“Human Phenotype Ontology Annotation and Cluster Analysis to Unravel Genetic Defects in 707 Cases with Unexplained Bleeding and Platelet Disorders.”</span> <em>Genome Medicine</em> 7 (1): 36. <a href="https://doi.org/10.1186/s13073-015-0151-5" class="external-link">https://doi.org/10.1186/s13073-015-0151-5</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vincent Carey.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
